--- a/cpp/lib/lib_mysql.cpp
+++ /dev/null
@@ -1,75 +0,0 @@
-// (C) AlgoEngineering LLC 2008-2013
-// (C) 2013-2018 NYSE | Intercontinental Exchange
-//
-// Target: lib_mysql (lib) -- Mysql adaptor
-// Exceptions: NO
-// Source:    cpp/gen/lib_mysql_gen.cpp
-// Source: -> cpp/lib/lib_mysql.cpp
-// Header:    include/gen/lib_mysql_gen.h
-// Header:    include/gen/lib_mysql_gen.inl.h
-// Header:    include/lib/lib_mysql.h
-//
-// Created By: alexei.lebedev
-// Recent Changes: alexei.lebedev
-//
-// Helper functions for mysql
-
-#include "include/lib/lib_mysql.h"
-#include "include/lib/lib_odbc.h"
-#include "include/lib/lib_iconv.h"
-
-// LHS      output buffer
-// CONN     mysql context (includes collation info)
-// RHS      string to print
-// QUOTES   quotes to use (' or "); if set to zero, quotes are omitted.
-void lib_mysql::MPrintQuoted(cstring &lhs, MYSQL *conn, strptr rhs, char quotes) {
-    strptr q;
-    if (quotes) {
-        q = strptr(&quotes,1);
-    }
-    lhs << q;
-    ch_Reserve(lhs, elems_N(rhs)*2+1);
-    u32 ret = mysql_real_escape_string(conn, lhs.ch_elems + lhs.ch_n, rhs.elems, elems_N(rhs));
-    vrfy(ret < lhs.ch_max-ch_N(lhs), "buffer overflow");
-    lhs.ch_n = ch_N(lhs) + ret;
-    lhs << q;
-}
-
-// replace first occurence of ? in LHS with Y
-// QUOTES   quotes to use (' or "); if set to zero, quotes are omitted.
-void lib_mysql::MBind(MYSQL *conn, cstring &lhs, strptr y, char quotes) {
-    i32_Range R = ch_FindFirst(lhs,'?');
-    vrfy(R.end > R.beg,tempstr()<< "No unbound arguments left in ["<<lhs<<"]");
-    tempstr tmp;
-    tmp << ch_FirstN(lhs,R.beg);
-    MPrintQuoted(tmp, conn, y, quotes);
-    tmp << ch_RestFrom(lhs,R.end);
-    lhs = tmp;
-}
-
-// Execute query and access result
-// Execute query and access result
-// Previous result, if present, is discarded.
-// If this is not done, you get
-//   "Commands out of sync; you can't run this command now"
-// error
-void lib_mysql::MQuery(MYSQL *conn, strptr query, lib_mysql::Res &res) {
-    verblog(query);
-    res_Cleanup(res);
-    vrfy(mysql_query(conn, Zeroterm(tempstr() << query))==0, mysql_error(conn));
-    res.res = mysql_use_result(conn);
-}
-
-void lib_mysql::mysql_Cleanup() {
-    if (lib_mysql::_db.mysql) {
-        mysql_close(lib_mysql::_db.mysql);
-        lib_mysql::_db.mysql = NULL;
-    }
-}
-
-void lib_mysql::res_Cleanup(lib_mysql::Res &res) {
-    if (res.res) {
-        mysql_free_result(res.res);
-        res.res = NULL;
-    }
-}
--- a/cpp/mysql2ssim.cpp
+++ /dev/null
@@ -1,389 +0,0 @@
-// (C) AlgoEngineering LLC 2008-2013
-// (C) 2013-2018 NYSE | Intercontinental Exchange
-//
-// Target: mysql2ssim (exe) -- mysql -> ssim conversion tool
-// Exceptions: NO
-// Source:    cpp/gen/mysql2ssim_gen.cpp
-// Source: -> cpp/mysql2ssim.cpp
-// Header:    include/gen/mysql2ssim_gen.h
-// Header:    include/gen/mysql2ssim_gen.inl.h
-//
-// Created By: alexei.lebedev alexey.polovinkin
-// Authors: alexei.lebedev
-// Recent Changes: alexei.lebedev alexey.polovinkin
-//
-// Dump contents of a mysql database, or some set of tables as an ssim file.
-// Optionally dump just the database schema.
-
-#include "include/gen/mysql2ssim_gen.h"
-#include "include/gen/mysql2ssim_gen.inl.h"
-#include "include/lib/lib_mysql.h"
-#include <mysql/mysqld_error.h> // for ER_BAD_DB_ERROR
-
-// -----------------------------------------------------------------------------
-
-static dmmeta::CtypePkey MapDecimal(mysql2ssim::FTobltin &tobltin, strptr param) {
-    i32_Range DR = TFind(param,',');
-    i32 fxsize;
-    i32 flsize;
-    dmmeta::CtypePkey ret;
-
-    vrfy(i32_ReadStrptrMaybe(fxsize, FirstN(param,DR.beg)), algo_lib::_db.errtext);
-    vrfy(i32_ReadStrptrMaybe(flsize, RestFrom(param,DR.end)), algo_lib::_db.errtext);
-
-    // i32 can store 9 digits, i64 - 18 digits (signed)
-    if (flsize == 0) {
-        ret = "i64";
-        tobltin.warn = true;
-    } else if ((fxsize+flsize < 9) && (flsize <= 4)) {
-        ret = tempstr() << "I32Dec" << flsize;
-    } else {
-        if (flsize > 7) {
-            tobltin.warn = true; // Precision will be limited
-            flsize = 7;
-        }
-        ret = tempstr() << "I64Dec" << flsize;
-    }
-    return ret;
-}
-
-// -----------------------------------------------------------------------------
-
-static void CvtEnum(mysql2ssim::FTobltin &tobltin, dmmeta::Field &field) {
-    int max = 0;
-    frep_(i,vals_N(tobltin)) {
-        max = i32_Max(max, ch_N(vals_qFind(tobltin,i)));
-    }
-    if (   (vals_qFind(tobltin,0)=="'N'" && vals_qFind(tobltin,1)=="'Y'")
-           || (vals_qFind(tobltin,0)=="'Y'" && vals_qFind(tobltin,1)=="'N'")) {
-        field.arg = "Bool";
-        field.dflt.value        = "false";
-    } else {
-        tobltin.warn = true;
-        field.arg = tempstr() << "Smallstr" << max;
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static void CvtChar(mysql2ssim::FTobltin &tobltin, dmmeta::Field &field) {
-    vrfy(vals_N(tobltin), "no values defined");
-    int width;
-    vrfy(i32_ReadStrptrMaybe(width, vals_qFind(tobltin,0)), algo_lib::_db.errtext);
-    field.arg = width == 1 ? strptr("char") : strptr(tempstr() << "RspaceStr"<<width);
-    field.dflt.value  = "' '";
-}
-
-// -----------------------------------------------------------------------------
-
-static void CvtVarchar(mysql2ssim::FTobltin &tobltin, dmmeta::Field &field) {
-    vrfy(vals_N(tobltin), "no values defined");
-    int  width;
-    vrfy(i32_ReadStrptrMaybe(width, vals_qFind(tobltin,0)), algo_lib::_db.errtext);
-    field.arg = width<=253 ? strptr(tempstr() << "Smallstr" << width) : strptr("algo.cstring");
-}
-
-// -----------------------------------------------------------------------------
-
-//
-// STR          input MYSQL type
-// FIELD        target TFC "field" record
-// Convert MYSQL type as best we can to a TFC type.
-//
-static void ToBltin(strptr str, dmmeta::Field &field) {
-    i32_Range  L      = TFind(str, '(');
-    i32_Range  R      = TFind(str, ')');
-    strptr type   = FirstN(str, L.beg);
-    strptr suffix = Trimmed(RestFrom(str, R.end));
-    bool   uns    = range_N(substr_FindFirst(suffix, strptr("unsigned"))) > 0;
-    strptr param  = qGetRegion(str, L.end, R.beg-L.end);
-    mysql2ssim::FTobltin  tobltin;
-    StringIter iter(param);
-    while (!iter.EofQ()) {
-        vals_Alloc(tobltin) = GetTokenChar(iter, ',');
-    }
-    if (type == "tinyint") {
-        field.arg = uns ? "u8" : "i8";
-    } else if (type == "mediumint" || type == "int") {
-        field.arg = uns ? "u32" : "i32";
-    } else if (type == "smallint") {
-        field.arg = uns ? "u16" : "i16";
-    } else if (type == "bigint") {
-        field.arg = uns ? "u64" : "i64";
-    } else if (type == "decimal" || type == "numeric") {
-        field.arg = MapDecimal(tobltin,param);
-    } else if (type == "float") {
-        field.arg = "float";
-        tobltin.warn = vals_N(tobltin) > 0;
-    } else if (type == "text") {
-        field.arg = "algo.cstring";
-    } else if (type == "double") {
-        field.arg = "double";
-        tobltin.warn = vals_N(tobltin) > 0;
-    } else if (type == "char") {
-        CvtChar(tobltin,field);
-    } else if (type == "varchar") {
-        CvtVarchar(tobltin,field);
-    } else if (type == "enum") {
-        CvtEnum(tobltin,field);
-    } else if (type == "datetime") {
-        field.arg = "UnTime";
-        field.dflt.value = "";
-    } else if (type == "date") {
-        field.arg = "UnixTime";
-        field.dflt.value = "";
-    } else if (type == "time") {
-        field.arg = "UnDiff";
-        field.dflt.value = "";
-    } else {
-        tobltin.err=true;
-    }
-    vrfy(!tobltin.err,tempstr()<< "cannot convert type "<<str);
-    if (tobltin.warn) {
-        prerr("mysql2ssim.conversion_warning"
-              <<Keyval("type",str)
-              <<Keyval("target",field.arg)
-              <<Keyval("comment","Cannot convert to target type"));
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static tempstr TableList() {
-    tempstr ret;
-    mysql2ssim::in_tables_RemoveAll();
-    strptr tables = mysql2ssim::_db.cmdline.tables;
-    while (elems_N(tables)) {
-        i32_Range r2   = TFind(tables,',');
-        mysql2ssim::in_tables_Alloc() = FirstN(tables,r2.beg);
-        tables = RestFrom(tables,r2.end);
-    }
-    ListSep ls(",");
-    ind_beg(mysql2ssim::_db_in_tables_curs, E, mysql2ssim::_db) {
-        ret << ls;
-        lib_mysql::MPrintQuoted(ret, lib_mysql::_db.mysql, E, '\'');
-    }ind_end;
-    return ret;
-}
-
-// -----------------------------------------------------------------------------
-
-static void LoadList(strptr db) {
-    cstring query;
-    query<<"SELECT table_name FROM information_schema.tables WHERE table_schema=?";
-    lib_mysql::MBind(lib_mysql::_db.mysql,query,db,'\'');
-    tempstr tables(TableList());
-    if (ch_N(tables)) {
-        query << " AND TABLE_NAME IN (" << tables << ")";
-    }
-    lib_mysql::Res    res;
-    MQuery(lib_mysql::_db.mysql, query, res);
-    MYSQL_ROW            row;
-    while ((row = mysql_fetch_row(res.res)) != NULL){
-        mysql2ssim::table_names_Alloc() = row[0];
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static void ReadSchema(strptr db, strptr table_name) {
-    tempstr ctype_out;
-    ctype_out << db;
-    ctype_out << ".";
-    strptr_PrintCamel(table_name, ctype_out);
-
-    dmmeta::Ctype ctype;
-    ctype.ctype  = ctype_out;
-    prlog(ctype);
-
-    // type came from database -- therefore link it back.
-    dmmeta::Ssimfile ssimfile;
-    ssimfile.ssimfile = tempstr() << db << "." << table_name;
-    ssimfile.ctype = ctype.ctype;
-    prlog(ssimfile);
-
-    cstring query;
-    query = "SELECT"
-        " table_schema AS `db`" // 0
-        " ,table_name AS `table`" // 1
-        " ,column_name AS `column`" // 2
-        " ,column_type AS `type`"   // 3
-        " ,is_nullable AS `null`"   // 4
-        " ,column_key AS `key`"     // 5
-        " ,column_default AS `dflt`" // 6
-        " FROM information_schema.columns where table_schema=? and table_name=?";
-    lib_mysql::MBind(lib_mysql::_db.mysql, query, db, '\'');
-    lib_mysql::MBind(lib_mysql::_db.mysql, query, table_name, '\'');
-
-    lib_mysql::Res res;
-    MQuery(lib_mysql::_db.mysql, query, res);
-    int    nfields       = mysql_field_count(lib_mysql::_db.mysql);
-    aryptr<MYSQL_FIELD>  fields(mysql_fetch_fields(res.res), nfields);
-    MYSQL_ROW            mysql_row;
-    int idx              = 0;
-
-    while ((mysql_row = mysql_fetch_row(res.res)) != NULL) {
-        prlog_("*");
-        dmmeta::Field field;
-        field.field         = tempstr() << ctype.ctype<<"." << mysql_row[2];
-        field.dflt.value  = mysql_row[6];
-        ToBltin(mysql_row[3], field);
-        idx++;
-        prlog(field);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static void ReadData_Ssim(Attr &head, lib_mysql::Res &res, int nfields, int skipfield, aryptr<MYSQL_FIELD> fields) {
-    cstring str;
-    MYSQL_ROW            mysql_row;
-    while ((mysql_row = mysql_fetch_row(res.res)) != NULL) {
-        PrintAttr(str, head.name, head.value);
-        frep_(i,nfields) {
-            if (mysql_row[i] && i!=skipfield) {
-                PrintAttrSpace(str, fields[i].name, mysql_row[i]);
-            }
-        }
-        if (mysql2ssim::_db.cmdline.writessimfile) {
-            str << eol;
-        } else {
-            //print string and flush
-            algo::Prlog(1/*stdout*/,str,0/*start index*/,true/*eol*/);
-        }
-    }
-    if (mysql2ssim::_db.cmdline.writessimfile) {
-        // append data to file
-        StringToFile(str, mysql2ssim::_db.tempfile.filename, algo_FileFlags_append);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static void ReadData_Table(Attr &head, lib_mysql::Res &res, int nfields, aryptr<MYSQL_FIELD> fields) {
-    // print aligned rectangular blocks
-    int screenful = 100;
-    algo_lib::FTxttbl table;
-    MYSQL_ROW            mysql_row;
-    while ((mysql_row = mysql_fetch_row(res.res)) != NULL) {
-        AddRow(table);
-        PrintAttr(AddCell(table), head.name, head.value);
-        frep_(i,nfields) {
-            cstring &cell = AddCell(table);
-            if (!mysql_row[i]) {
-                continue;
-            }
-            PrintAttr(cell, fields[i].name, mysql_row[i]);
-        }
-        if (c_txtrow_N(table) >= screenful) {
-            prlog(table);
-            Refurbish(table);
-        }
-    }
-    if (c_txtrow_N(table) > 0) {
-        prlog(table);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static void ReadData(strptr db, strptr table_name) {
-    cstring query("SELECT * FROM ?");
-    Attr head;
-    lib_mysql::MBind(lib_mysql::_db.mysql, query, table_name, '`');
-    ch_RemoveAll(head.value);
-    head.value<< db << "." << table_name;
-    lib_mysql::Res res;
-    lib_mysql::MQuery(lib_mysql::_db.mysql, query, res);
-    int    nfields       = mysql_field_count(lib_mysql::_db.mysql);
-    aryptr<MYSQL_FIELD>  fields(mysql_fetch_fields(res.res), nfields);
-    // on the way back, skip the bad column
-    int skipfield=-1;
-    frep_(i,nfields) {
-        if (strptr(fields[i].name) == strptr("extra_column_for_roundtrip")) {
-            skipfield=i;
-        }
-    }
-    //
-    // print tuples
-    //
-    if (!mysql2ssim::_db.cmdline.pretty) {
-        ReadData_Ssim(head,res,nfields,skipfield,fields);
-    } else {
-        ReadData_Table(head,res,nfields,fields);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static void Logo() {
-    // identify ourselves
-    cstring revision;
-    revision << "# (C) 2013-2014 NYSE|Intercontinental Exchange\n";
-    revision << "# generated by mysql2ssim $Revision:: 2519$\n";
-    Replace(revision, "$", "");
-    if (!mysql2ssim::_db.cmdline.nologo) {
-        prlog(revision);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static void Connect(URL &url, strptr db) {
-    strptr sock = Pathcomp(url.dir,"/RL");
-    bool issock = url.protocol == "sock";
-    bool connect_ok = mysql_real_connect(lib_mysql::_db.mysql
-                                         , (issock ? NULL : Zeroterm(tempstr() << url.server))
-                                         , Zeroterm(tempstr() << url.username)
-                                         , Zeroterm(tempstr() << url.password)
-                                         , Zeroterm(tempstr() << db)
-                                         , 0
-                                         , (issock ? Zeroterm(tempstr() << sock) : NULL)
-                                         , 0);
-    if (!connect_ok && ER_BAD_DB_ERROR == mysql_errno(lib_mysql::_db.mysql) && mysql2ssim::_db.cmdline.baddbok) {
-        // quietly finish with success, nothing to do
-        _exit(0);
-    }
-    vrfy(connect_ok,mysql_error(lib_mysql::_db.mysql));
-}
-
-// -----------------------------------------------------------------------------
-
-void mysql2ssim::Main() {
-    lib_mysql::_db.mysql = mysql_init(NULL);
-
-    if (mysql2ssim::_db.cmdline.writessimfile) {
-        TempfileInitX(mysql2ssim::_db.tempfile,"mysql2ssim");
-    }
-
-
-    URL url;
-    vrfy(URL_ReadStrptrMaybe(url, mysql2ssim::_db.cmdline.url), algo_lib::_db.errtext);
-
-    Logo();
-    strptr db = Pathcomp(url.dir, "/RR");
-    Connect(url,db);
-
-    // load list of tables
-    LoadList(db);
-
-    ind_beg(mysql2ssim::_db_table_names_curs, table_name, mysql2ssim::_db) {
-        if (mysql2ssim::_db.cmdline.schema) {
-            ReadSchema(db,table_name);
-        } else {
-            ReadData(db,table_name);
-        }
-    }ind_end;
-
-    // use acr to append this data to tables
-    if (mysql2ssim::_db.cmdline.writessimfile) {
-        command::acr acr;
-        acr.trunc=true;
-        acr.replace=true;
-        acr.write=true;
-        acr.report=false;
-        acr.print=false;
-        int rc=SysCmd(acr_ToCmdline(acr)<<"< "<<mysql2ssim::_db.tempfile.filename);
-        algo_lib::_db.exit_code += Abs(rc);
-    }
-}
--- a/cpp/ssim2mysql.cpp
+++ /dev/null
@@ -1,376 +0,0 @@
-// (C) AlgoEngineering LLC 2008-2013
-// (C) 2013-2018 NYSE | Intercontinental Exchange
-//
-// Target: ssim2mysql (exe) -- Ssim -> mysql
-// Exceptions: yes
-// Source:    cpp/gen/ssim2mysql_gen.cpp
-// Source: -> cpp/ssim2mysql.cpp
-// Header:    include/gen/ssim2mysql_gen.h
-// Header:    include/gen/ssim2mysql_gen.inl.h
-//
-// Created By: alexei.lebedev alexey.polovinkin
-// Authors: alexei.lebedev
-// Recent Changes: alexei.lebedev anand.pradhan alexey.polovinkin
-//
-// Convert SSIM data to SQL.
-// Print SQL to the screen to submit to mysql
-
-#include "include/lib/lib_mysql.h"
-#include "include/gen/ssim2mysql_gen.h"
-#include "include/gen/ssim2mysql_gen.inl.h"
-
-
-static void NewCmd(strptr text) {
-    ssim2mysql::FCmd &cmd = ssim2mysql::cmd_Alloc();
-    cmd.cmd << text;
-    vrfy(cmd_XrefMaybe(cmd), algo_lib::_db.errtext);
-    verblog("Cmd: "<<text);
-}
-
-static void ProcessLine_Ssimfile(ssim2mysql::FInput *input, ssim2mysql::FSsimfile *ssimfile) {
-    // check if tuple contains attributes which are not in the database.
-    // warn.
-    ind_beg(algo::Tuple_attrs_curs, attr, input->tuple) {
-        tempstr colname;
-        colname << ssimfile->ssimfile << "." << attr.name;
-        if (!ssim2mysql::ind_column_Find(colname)) {
-            tempstr tmps;
-            tmps << "WARNING: Column [" << colname << "] not found and won't be added to MySQL table";
-            if (tmps != input->warnstr) {
-                prerr(tmps);
-                input->warnstr = tmps;
-            }
-        }
-    }ind_end;
-    ListSep field_sep(",");
-    if (ssim2mysql::_db.n_cmd_rows > 0) {
-        ssim2mysql::_db.c_cmd_cur->cmd << ",\n";
-    }
-    ssim2mysql::_db.n_cmd_rows++;
-    ssim2mysql::_db.c_cmd_cur->cmd << "(" ;
-    ind_beg(ssim2mysql::ssimfile_c_column_curs, column,*ssimfile) {
-        strptr val;
-        if (column.p_field->c_substr) {
-            val = attr_GetString(input->tuple, StripNs("",column.p_field->c_substr->srcfield), column.defval);
-            val = Pathcomp(val, column.p_field->c_substr->expr.value);
-        } else {
-            val = attr_GetString(input->tuple, name_Get(column), column.defval);
-        }
-        ssim2mysql::_db.c_cmd_cur->cmd << field_sep;
-        if (elems_N(val) && column.is_int) {
-            ssim2mysql::_db.c_cmd_cur->cmd << val;
-        } else if (ssim2mysql::_db.cmdline.dry_run) {
-            strptr_PrintCppQuoted(val, ssim2mysql::_db.c_cmd_cur->cmd, '\''); // some escaping
-        } else {
-            lib_mysql::MPrintQuoted(ssim2mysql::_db.c_cmd_cur->cmd,lib_mysql::_db.mysql,val,'\'');
-        }
-    }ind_end;
-    ssim2mysql::_db.c_cmd_cur->cmd << ")";
-}
-
-//
-// Add input tuple to current query (FCmd)
-// Empty line flushes the query (submits for execution)
-//
-static void ProcessLine(ssim2mysql::FInput *input, strptr line) {
-    vrfy(Tuple_ReadStrptrMaybe(input->tuple,line), algo_lib::_db.errtext);
-    strptr table = input->tuple.head.value;
-    ssim2mysql::FSsimfile *ssimfile = ssim2mysql::ind_ssimfile_Find(table);
-
-    // table changed? end old insert statement, switch current ssimfile
-    bool change = ssim2mysql::_db.c_ssimfile_cur != ssimfile;
-    change |= (ssim2mysql::_db.c_cmd_cur && ch_N(ssim2mysql::_db.c_cmd_cur->cmd) >= ssim2mysql::_db.cmdline.maxpacket *3/4);
-
-    if (ssim2mysql::_db.c_cmd_cur && change) {
-        ssim2mysql::_db.c_cmd_cur->cmd  << ";\n";
-        vrfy(cmd_XrefMaybe(*ssim2mysql::_db.c_cmd_cur), algo_lib::_db.errtext);// execute!
-        ssim2mysql::_db.c_cmd_cur = NULL;
-        ssim2mysql::_db.n_cmd_rows = 0;
-    }
-
-    ssim2mysql::_db.c_ssimfile_cur = ssimfile;
-
-    // start new insert command
-    if (ssimfile && !ssim2mysql::_db.c_cmd_cur) {
-        tempstr sql_table;
-        sql_table << "`" << GetNs(ssimfile->ssimfile) << "`.`" << StripNs("",table) << "`";
-
-        // if this is the first row for a table, and truncation was specified on command line,
-        // truncate.
-        if (ssim2mysql::_db.cmdline.trunc && !ssimfile->truncated) {
-            ssimfile->truncated = true;
-            NewCmd(tempstr() << "TRUNCATE TABLE "<<sql_table<<";\n");
-        }
-
-        ssim2mysql::FCmd &cmd = ssim2mysql::cmd_Alloc();
-        ssim2mysql::_db.c_cmd_cur = &cmd;
-        cmd.cmd << (ssim2mysql::_db.cmdline.replace ? "REPLACE":"INSERT") << " INTO "<<sql_table;        // begin insert statement
-
-        ListSep field_sep(",");// build column list
-        cmd.cmd <<" (";
-        ind_beg(ssim2mysql::ssimfile_c_column_curs, column,*ssimfile) {
-            cmd.cmd << field_sep << "`" << name_Get(column) << "`";
-        }ind_end;
-        cmd.cmd <<")";
-        cmd.cmd <<" VALUES\n" ;
-    }
-
-    if (ssimfile) {
-        ProcessLine_Ssimfile(input, ssimfile);
-    }
-}
-
-static void EofActions() {
-    if (ssim2mysql::_db.c_cmd_cur) {
-        ssim2mysql::_db.c_cmd_cur->cmd  << ";\n";
-        vrfy(cmd_XrefMaybe(*ssim2mysql::_db.c_cmd_cur), algo_lib::_db.errtext);// execute!
-        ssim2mysql::_db.c_cmd_cur = NULL;
-    }
-
-    if (ssim2mysql::_db.cmdline.createdb && ssim2mysql::_db.cmdline.fkey) {     // Re-create database
-        ssim2mysql::FNs &fns = ssim2mysql::ind_ns_FindX(ssim2mysql::_db.cmdline.db);
-
-        // create foreign key constraints
-        ind_beg(ssim2mysql::ns_c_ssimfile_curs, ssimfile, fns) {
-            ind_beg(ssim2mysql::ctype_c_field_curs, field, *ssimfile.p_ctype) if (field.select && field.is_pkeyref) { // field, e.g. acmdb.Devchassis.chassis
-                ssim2mysql::FSsimfile *tgt_ssimfile = field.p_arg->c_ssimfile;// target ssimfile, e.g. acmdb.chassis
-                ssim2mysql::FField *tgt_field = c_field_Find(*field.p_arg, 0); // first field of target ctype, e.g. acmdb.Chassis.chassis
-                if (tgt_ssimfile && zd_ssimfile_InLlistQ(*tgt_ssimfile)) {
-                    NewCmd(tempstr()
-                           << "ALTER TABLE `"<< name_Get(ssimfile)<<"`"                                              // e.g. devchassis
-                           <<" ADD CONSTRAINT fk_" << name_Get(ssimfile) << "_" << name_Get(field)             // e.g. devchassis_chassis
-                           << " FOREIGN KEY (`" << name_Get(field) << "`)"                                       // e.g. chassis
-                           << " REFERENCES `" << name_Get(*tgt_ssimfile) << "` (`"<< name_Get(*tgt_field) <<"`);"); // e.g. chassis_chassis
-                }
-            } ind_end;
-        }ind_end;
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-//
-// read lines out of input buffer.
-// do not process partial lines (they could be corrupted)
-//
-void ssim2mysql::cd_input_line_Step() {
-    ssim2mysql::FInput &input = *ssim2mysql::cd_input_line_RotateFirst();
-    aryptr<char> line    = in_buf_GetMsg(input);
-    if (line.elems != NULL) {
-        ProcessLine(&input, line);
-        in_buf_SkipMsg(input);
-    } else if (input.in_buf_eof) {
-        EofActions();// end-of-file -- apply constraints  etc.
-        if (input.in_buf_err.value != 0) {
-            prerr("ssim2mysql.input"
-                  <<Keyval("error",input.in_buf_err));
-            algo_lib::_db.exit_code = 1;
-        }
-        input_Delete(input);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-static ssim2mysql::FField *RootField(ssim2mysql::FField *root_fld) {
-    do {
-        ssim2mysql::FCtype *ft = root_fld->p_arg;
-        if (ft->c_sqltype || c_field_N(*ft) == 0) {
-            break;
-        }
-        root_fld = c_field_Find(*ft, 0);
-    } while (true);
-    return root_fld;
-}
-
-// -----------------------------------------------------------------------------
-
-static tempstr Dflt(ssim2mysql::FField *root_fld) {
-    ssim2mysql::FSqltype *sqltype = root_fld->p_arg->c_sqltype; // builtin?
-    bool istime = FindStr(sqltype->expr, strptr("time")) != -1;
-    bool isstring = FindStr(sqltype->expr, strptr("char")) != -1 || sqltype->expr == "text";
-    tempstr dflt;
-    if (root_fld->arg == "Bool" || root_fld->arg == "bool") {
-        dflt << (root_fld->dflt.value == "true" ? "'Y'" : "'N'");
-    } else if (isstring || (istime && ("" != root_fld->dflt.value))) {
-        // the default value is a c++ expression.
-        // since c++ escaping is close enough to sql escaping, we will just print the string as-is,
-        dflt << root_fld->dflt;
-        if (!ch_N(dflt)) {
-            dflt << "''";
-        }
-    } else if ("" != root_fld->dflt.value) {   // Numerical defaults can not be empty strings
-        dflt << root_fld->dflt;
-    } else {
-        dflt << "0";// numeric
-    }
-    return dflt;
-}
-
-// -----------------------------------------------------------------------------
-
-static bool GoodReftype(dmmeta::ReftypeId reftype) {
-    return reftype == dmmeta_ReftypeId_Pkey
-        || reftype == dmmeta_ReftypeId_Val
-        || reftype == dmmeta_ReftypeId_RegxSql;// also a string
-}
-
-// -----------------------------------------------------------------------------
-
-static void FixupFields(ssim2mysql::FCtype &ctype) {
-    int n_field = 0;
-    ind_beg(ssim2mysql::ctype_c_field_curs, fld, ctype) {
-        bool good = !fld.c_substr || ssim2mysql::_db.cmdline.fldfunc;
-        n_field += good;
-    }ind_end;
-
-    // a table with a single column will get sorted by MySQL.
-    // must add another column to prevent this
-    // a ctype with no fields gets a single field.
-    if (n_field < 2) {
-        vrfy(ssim2mysql::field_InsertMaybe(dmmeta::Field(tempstr()<<ctype.ctype<<".extra_column_for_roundtrip"
-                                                         ,"algo.Comment"
-                                                         , "Val"
-                                                         ,dmmeta::CppExpr()
-                                                         , algo::Comment())), algo_lib::_db.errtext);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-//
-// Generate SQL command to create a table for given ssimfile
-//
-static void Main_GenSchema_Table(ssim2mysql::FSsimfile &ssimfile) {
-    zd_ssimfile_Insert(ssimfile); // table to upload
-    strptr engine = ssim2mysql::_db.cmdline.fkey ? "InnoDB" : "MyISAM";
-    bool surrogate_pkey = engine == "InnoDB";
-    ssim2mysql::FCtype &ctype = *ssimfile.p_ctype;
-    NewCmd(tempstr() << "DROP TABLE IF EXISTS `" << name_Get(ssimfile) << "`;");
-    ssim2mysql::FCmd &createcmd = ssim2mysql::cmd_Alloc();
-    createcmd.cmd << "CREATE TABLE `" << name_Get(ssimfile) << "` (\n";
-
-    FixupFields(ctype);
-    ListSep ls("\n, ");
-
-    if (surrogate_pkey) {
-        createcmd.cmd << ls;
-        createcmd.cmd << " `"<<name_Get(ssimfile) << "_rowid` INT PRIMARY KEY AUTO_INCREMENT";
-    }
-
-    // walk over all columns
-    ind_beg(ssim2mysql::ctype_c_field_curs, fld, ctype) {
-        // determine if the column should appear.
-        // fldfuncs are excluded unless explicitly requested
-        // (they cannot show up in the web-oriented database, and they
-        // do not make it back via mysql2ssim, but they simplify writing of joins)
-        bool good = !fld.c_substr || ssim2mysql::_db.cmdline.fldfunc;
-        fld.select = good;
-        //fld.is_pkeyref = fld.reftype == dmmeta_Reftype_reftype_Pkey;
-        dmmeta::ReftypeId reftype_id;
-        value_SetStrptr(reftype_id,fld.reftype,dmmeta_ReftypeId_Val);
-        dmmeta_ReftypeIdEnum reftype = value_GetEnum(reftype_id);
-        fld.is_pkeyref = reftype == dmmeta_ReftypeId_Pkey;
-        if (good && GoodReftype(reftype)) {
-            createcmd.cmd << ls;
-            createcmd.cmd << "`" << name_Get(fld) << "` ";       // Column name
-            ssim2mysql::FField *root_fld = RootField(&fld);// follow pkey chain to find out true type.
-            dmmeta::ReftypeId root_reftype_id;
-            value_SetStrptr(root_reftype_id,root_fld->reftype,dmmeta_ReftypeId_Val);
-            ssim2mysql::FSqltype *sqltype = root_fld->p_arg->c_sqltype; // builtin?
-            vrfy(sqltype, tempstr()<<"no sqltype defined for ctype "<<root_fld->arg);
-            createcmd.cmd << sqltype->expr;
-            tempstr dflt=Dflt(root_fld);
-            // Default value
-            createcmd.cmd << " NOT NULL ";
-            if (ch_N(dflt)) createcmd.cmd << " DEFAULT "<<dflt;
-            if (ind_curs(fld).index == 0) {
-                createcmd.cmd << (surrogate_pkey ? " UNIQUE" : " PRIMARY");
-                createcmd.cmd << " KEY ";
-            }
-        }
-    }ind_end;
-    createcmd.cmd << "\n) ENGINE="<< engine
-                  <<" DEFAULT CHARSET=latin1 COLLATE latin1_general_cs;\n\n";
-    vrfy(ssim2mysql::cmd_XrefMaybe(createcmd), algo_lib::_db.errtext);
-}
-
-// -----------------------------------------------------------------------------
-
-void ssim2mysql::zs_cmd_Step() {
-    // execute some commands
-    for(int iter = 0; ssim2mysql::zs_cmd_First() && iter < 5; iter++) {
-        ssim2mysql::FCmd &exec_cmd = *ssim2mysql::zs_cmd_RemoveFirst();
-        if (ssim2mysql::_db.cmdline.dry_run) {
-            prlog(exec_cmd.cmd);
-        }
-        verblog(exec_cmd.cmd);
-        if (!ssim2mysql::_db.cmdline.dry_run) {
-            vrfy(mysql_query(lib_mysql::_db.mysql, Zeroterm(exec_cmd.cmd))==0
-                 , mysql_error(lib_mysql::_db.mysql));
-        }
-        ssim2mysql::cmd_Delete(exec_cmd);
-    }
-}
-
-// -----------------------------------------------------------------------------
-
-void ssim2mysql::Main() {
-    URL url;
-    vrfy(URL_ReadStrptrMaybe(url, ssim2mysql::_db.cmdline.url), algo_lib::_db.errtext);
-    if (!ch_N(ssim2mysql::_db.cmdline.url)) {
-        ssim2mysql::_db.cmdline.dry_run = true;
-    }
-
-    // inputect to database
-    // NOTE empty server name defaults to localhost, so failure to supply
-    // server causes inputectino to wrong server
-    // During inputection, omit database name since it may not exist
-    bool   issock = url.protocol == "sock";
-    if (!ssim2mysql::_db.cmdline.dry_run) {
-        lib_mysql::_db.mysql = mysql_init(NULL);
-        vrfy(issock || ch_N(url.server)>0,tempstr()<< "server name must be specified");
-        vrfy(mysql_real_connect(lib_mysql::_db.mysql
-                                , (issock ? NULL : Zeroterm(tempstr() << url.server))
-                                , (issock ? NULL : Zeroterm(tempstr() << url.username))
-                                , (issock ? NULL : Zeroterm(tempstr() << url.password))
-                                , NULL // omit database name
-                                , 0
-                                , (issock ? Zeroterm(tempstr() << url.dir) : NULL)
-                                , 0)
-             , mysql_error(lib_mysql::_db.mysql));
-    }
-
-    // Re-create database
-    if (ssim2mysql::_db.cmdline.createdb) {
-        ssim2mysql::FNs &fns = ssim2mysql::ind_ns_FindX(ssim2mysql::_db.cmdline.db);
-        NewCmd(tempstr() << "DROP DATABASE IF EXISTS `"<<ssim2mysql::_db.cmdline.db<<"`;");
-        NewCmd(tempstr() << "CREATE DATABASE `"<<ssim2mysql::_db.cmdline.db<<"` DEFAULT CHARACTER SET latin1 COLLATE latin1_general_cs;");
-        NewCmd(tempstr() << "USE `"<<ssim2mysql::_db.cmdline.db<<"`;");
-
-        ind_beg(ssim2mysql::ns_c_ssimfile_curs, ssimfile,fns) {
-            Main_GenSchema_Table(ssimfile);
-        } ind_end;
-    }
-
-    // Create columns and fill in their properties
-    ind_beg(ssim2mysql::_db_ssimfile_curs, ssimfile, ssim2mysql::_db) {
-        ssim2mysql::FCtype &ctype = *ssimfile.p_ctype;
-        ind_beg(ssim2mysql::ctype_c_field_curs, field, ctype) if (!field.c_substr || ssim2mysql::_db.cmdline.fldfunc) {
-            tempstr colkey;
-            colkey << ssimfile.ssimfile << "." << name_Get(field);
-            ssim2mysql::FColumn   &col     = ssim2mysql::ind_column_GetOrCreate(colkey);
-            col.p_field     = &field;
-            col.defval      = strptr(field.dflt.value);
-            col.is_int      = ctype.c_sqltype ? range_N(substr_FindFirst(ctype.c_sqltype->expr, "int"))>0 : false;
-        }ind_end;
-    }ind_end;
-
-    // set up reading -- make sure to set nonblocking mode
-    ssim2mysql::FInput &input = ssim2mysql::input_Alloc();
-    algo::Fildes fildes((ssim2mysql::_db.cmdline.in=="-") ? Fildes(dup(0)) : OpenRead(ssim2mysql::_db.cmdline.in));
-    (void)SetBlockingMode(fildes, false);
-    in_buf_BeginRead(input, fildes);
-
-    ssim2mysql::MainLoop();
-}
--- a/include/lib/lib_mysql.h
+++ /dev/null
@@ -1,53 +0,0 @@
-// (C) AlgoEngineering LLC 2008-2013
-// (C) 2013-2018 NYSE | Intercontinental Exchange
-//
-// Target: lib_mysql (lib) -- Mysql adaptor
-// Exceptions: NO
-// Source:    cpp/gen/lib_mysql_gen.cpp
-// Source:    cpp/lib/lib_mysql.cpp
-// Header:    include/gen/lib_mysql_gen.h
-// Header:    include/gen/lib_mysql_gen.inl.h
-// Header: -> include/lib/lib_mysql.h
-//
-// Created By: alexei.lebedev
-// Recent Changes: alexei.lebedev
-//
-
-#pragma once
-#include "include/algo.h"
-#include "include/gen/lib_mysql_gen.h"
-#include "include/gen/lib_mysql_gen.inl.h"
-typedef struct st_mysql MYSQL;
-typedef struct st_mysql_res MYSQL_RES;
-
-namespace lib_mysql { // update-hdr
-    // Dear human:
-    //     Text from here to the closing curly brace was produced by scanning
-    //     source files. Editing this text is futile.
-    //     To refresh the contents of this section, run 'update-hdr'.
-    //     To convert this section to a hand-written section, remove the word 'update-hdr' from namespace line.
-
-    // -------------------------------------------------------------------
-    // cpp/lib/lib_mysql.cpp
-    //
-
-    // LHS      output buffer
-    // CONN     mysql context (includes collation info)
-    // RHS      string to print
-    // QUOTES   quotes to use (' or "); if set to zero, quotes are omitted.
-    void MPrintQuoted(cstring &lhs, MYSQL *conn, strptr rhs, char quotes);
-
-    // replace first occurence of ? in LHS with Y
-    // QUOTES   quotes to use (' or "); if set to zero, quotes are omitted.
-    void MBind(MYSQL *conn, cstring &lhs, strptr y, char quotes);
-
-    // Execute query and access result
-    // Execute query and access result
-    // Previous result, if present, is discarded.
-    // If this is not done, you get
-    // "Commands out of sync; you can't run this command now"
-    // error
-    void MQuery(MYSQL *conn, strptr query, lib_mysql::Res &res);
-    void mysql_Cleanup();
-    void res_Cleanup(lib_mysql::Res &res);
-}
--- a/data/dev/gitfile.ssim
+++ b/data/dev/gitfile.ssim
@@ -259,18 +259,15 @@ dev.gitfile  gitfile:cpp/gen/lib_exec_gen.cpp
 dev.gitfile  gitfile:cpp/gen/lib_git_gen.cpp
 dev.gitfile  gitfile:cpp/gen/lib_iconv_gen.cpp
 dev.gitfile  gitfile:cpp/gen/lib_json_gen.cpp
-dev.gitfile  gitfile:cpp/gen/lib_mysql_gen.cpp
 dev.gitfile  gitfile:cpp/gen/lib_odbc_gen.cpp
 dev.gitfile  gitfile:cpp/gen/lib_sql_gen.cpp
 dev.gitfile  gitfile:cpp/gen/lin_gen.cpp
-dev.gitfile  gitfile:cpp/gen/mysql2ssim_gen.cpp
 dev.gitfile  gitfile:cpp/gen/pcap_gen.cpp
 dev.gitfile  gitfile:cpp/gen/report_gen.cpp
 dev.gitfile  gitfile:cpp/gen/src_func_gen.cpp
 dev.gitfile  gitfile:cpp/gen/src_hdr_gen.cpp
 dev.gitfile  gitfile:cpp/gen/src_lim_gen.cpp
 dev.gitfile  gitfile:cpp/gen/ssim2csv_gen.cpp
-dev.gitfile  gitfile:cpp/gen/ssim2mysql_gen.cpp
 dev.gitfile  gitfile:cpp/gen/strconv_gen.cpp
 dev.gitfile  gitfile:cpp/lib/algo/arg.cpp
 dev.gitfile  gitfile:cpp/lib/algo/backtrace.cpp
@@ -298,10 +295,8 @@ dev.gitfile  gitfile:cpp/lib/lib_gen.cpp
 dev.gitfile  gitfile:cpp/lib/lib_git.cpp
 dev.gitfile  gitfile:cpp/lib/lib_iconv.cpp
 dev.gitfile  gitfile:cpp/lib/lib_json.cpp
-dev.gitfile  gitfile:cpp/lib/lib_mysql.cpp
 dev.gitfile  gitfile:cpp/lib/lib_odbc.cpp
 dev.gitfile  gitfile:cpp/lib/lib_sql.cpp
-dev.gitfile  gitfile:cpp/mysql2ssim.cpp
 dev.gitfile  gitfile:cpp/src/func/check.cpp
 dev.gitfile  gitfile:cpp/src/func/edit.cpp
 dev.gitfile  gitfile:cpp/src/func/fileloc.cpp
@@ -314,7 +309,6 @@ dev.gitfile  gitfile:cpp/src/func/scan.cpp
 dev.gitfile  gitfile:cpp/src/hdr.cpp
 dev.gitfile  gitfile:cpp/src/lim.cpp
 dev.gitfile  gitfile:cpp/ssim2csv.cpp
-dev.gitfile  gitfile:cpp/ssim2mysql.cpp
 dev.gitfile  gitfile:cpp/strconv/strconv.cpp
 dev.gitfile  gitfile:data/amcdb/bltin.ssim
 dev.gitfile  gitfile:data/amcdb/curs.ssim
@@ -558,16 +552,12 @@ dev.gitfile  gitfile:include/gen/lib_iconv_gen.h
 dev.gitfile  gitfile:include/gen/lib_iconv_gen.inl.h
 dev.gitfile  gitfile:include/gen/lib_json_gen.h
 dev.gitfile  gitfile:include/gen/lib_json_gen.inl.h
-dev.gitfile  gitfile:include/gen/lib_mysql_gen.h
-dev.gitfile  gitfile:include/gen/lib_mysql_gen.inl.h
 dev.gitfile  gitfile:include/gen/lib_odbc_gen.h
 dev.gitfile  gitfile:include/gen/lib_odbc_gen.inl.h
 dev.gitfile  gitfile:include/gen/lib_sql_gen.h
 dev.gitfile  gitfile:include/gen/lib_sql_gen.inl.h
 dev.gitfile  gitfile:include/gen/lin_gen.h
 dev.gitfile  gitfile:include/gen/lin_gen.inl.h
-dev.gitfile  gitfile:include/gen/mysql2ssim_gen.h
-dev.gitfile  gitfile:include/gen/mysql2ssim_gen.inl.h
 dev.gitfile  gitfile:include/gen/pcap_gen.h
 dev.gitfile  gitfile:include/gen/pcap_gen.inl.h
 dev.gitfile  gitfile:include/gen/report_gen.h
@@ -580,8 +570,6 @@ dev.gitfile  gitfile:include/gen/src_lim_gen.h
 dev.gitfile  gitfile:include/gen/src_lim_gen.inl.h
 dev.gitfile  gitfile:include/gen/ssim2csv_gen.h
 dev.gitfile  gitfile:include/gen/ssim2csv_gen.inl.h
-dev.gitfile  gitfile:include/gen/ssim2mysql_gen.h
-dev.gitfile  gitfile:include/gen/ssim2mysql_gen.inl.h
 dev.gitfile  gitfile:include/gen/strconv_gen.h
 dev.gitfile  gitfile:include/gen/strconv_gen.inl.h
 dev.gitfile  gitfile:include/lib/lib_ctype.h
@@ -589,7 +577,6 @@ dev.gitfile  gitfile:include/lib/lib_exec.h
 dev.gitfile  gitfile:include/lib/lib_iconv.h
 dev.gitfile  gitfile:include/lib/lib_json.h
 dev.gitfile  gitfile:include/lib/lib_json.inl.h
-dev.gitfile  gitfile:include/lib/lib_mysql.h
 dev.gitfile  gitfile:include/lib/lib_odbc.h
 dev.gitfile  gitfile:include/lib_git.h
 dev.gitfile  gitfile:include/lib_sql.h
--- a/data/dev/linelim.ssim
+++ b/data/dev/linelim.ssim
@@ -121,4 +121,3 @@ dev.linelim  gitfile:cpp/lib/lib_exec.cpp  nlongline:0  longestline:118  nbadws:
 dev.linelim  gitfile:cpp/lib/lib_json.cpp  nlongline:1  longestline:126  nbadws:0  maxws:16  nlongfunc:0  longestfunc:50  nmysteryfunc:7  badness:*****************
 dev.linelim  gitfile:cpp/lib/lib_odbc.cpp  nlongline:0  longestline:118  nbadws:0  maxws:12  nlongfunc:0  longestfunc:57  nmysteryfunc:13  badness:******************************
 dev.linelim  gitfile:cpp/src/lim.cpp  nlongline:2  longestline:145  nbadws:0  maxws:8  nlongfunc:0  longestfunc:53  nmysteryfunc:1  badness:.....
-dev.linelim  gitfile:cpp/ssim2mysql.cpp  nlongline:9  longestline:157  nbadws:0  maxws:16  nlongfunc:0  longestfunc:58  nmysteryfunc:2  badness:...........
--- a/data/dev/targdep.ssim
+++ b/data/dev/targdep.ssim
@@ -44,15 +44,11 @@ dev.targdep  targdep:lib_git.algo_pch  comment:""
 dev.targdep  targdep:lib_git.gen_lib  comment:""
 dev.targdep  targdep:lib_iconv.algo_pch  comment:""
 dev.targdep  targdep:lib_json.algo_pch  comment:""
-dev.targdep  targdep:lib_mysql.algo_pch  comment:""
 dev.targdep  targdep:lib_odbc.algo_pch  comment:""
 dev.targdep  targdep:lib_odbc.lib_iconv  comment:""
 dev.targdep  targdep:lib_sql.algo_lib  comment:""
 dev.targdep  targdep:lib_sql.algo_pch  comment:""
 dev.targdep  targdep:lib_sql.gen_lib  comment:""
-dev.targdep  targdep:mysql2ssim.algo_lib  comment:""
-dev.targdep  targdep:mysql2ssim.algo_pch  comment:""
-dev.targdep  targdep:mysql2ssim.lib_mysql  comment:""
 dev.targdep  targdep:src_func.algo_lib  comment:""
 dev.targdep  targdep:src_func.algo_pch  comment:""
 dev.targdep  targdep:src_func.gen_lib  comment:""
@@ -64,9 +60,6 @@ dev.targdep  targdep:src_lim.algo_pch  comment:""
 dev.targdep  targdep:src_lim.gen_lib  comment:""
 dev.targdep  targdep:ssim2csv.algo_lib  comment:""
 dev.targdep  targdep:ssim2csv.algo_pch  comment:""
-dev.targdep  targdep:ssim2mysql.algo_lib  comment:""
-dev.targdep  targdep:ssim2mysql.algo_pch  comment:""
-dev.targdep  targdep:ssim2mysql.lib_mysql  comment:""
 dev.targdep  targdep:strconv.algo_lib  comment:""
 dev.targdep  targdep:strconv.algo_pch  comment:""
 dev.targdep  targdep:strconv.gen_lib  comment:""
--- a/data/dev/target.ssim
+++ b/data/dev/target.ssim
@@ -19,13 +19,10 @@ dev.target  target:lib_exec
 dev.target  target:lib_git
 dev.target  target:lib_iconv
 dev.target  target:lib_json
-dev.target  target:lib_mysql
 dev.target  target:lib_odbc
 dev.target  target:lib_sql
-dev.target  target:mysql2ssim
 dev.target  target:src_func
 dev.target  target:src_hdr
 dev.target  target:src_lim
 dev.target  target:ssim2csv
-dev.target  target:ssim2mysql
 dev.target  target:strconv
--- a/data/dev/targsrc.ssim
+++ b/data/dev/targsrc.ssim
@@ -311,11 +311,6 @@ dev.targsrc  targsrc:lib_json/include/gen/lib_json_gen.h  comment:""
 dev.targsrc  targsrc:lib_json/include/gen/lib_json_gen.inl.h  comment:""
 dev.targsrc  targsrc:lib_json/include/lib/lib_json.h  comment:""
 dev.targsrc  targsrc:lib_json/include/lib/lib_json.inl.h  comment:""
-dev.targsrc  targsrc:lib_mysql/cpp/gen/lib_mysql_gen.cpp  comment:""
-dev.targsrc  targsrc:lib_mysql/cpp/lib/lib_mysql.cpp  comment:""
-dev.targsrc  targsrc:lib_mysql/include/gen/lib_mysql_gen.h  comment:""
-dev.targsrc  targsrc:lib_mysql/include/gen/lib_mysql_gen.inl.h  comment:""
-dev.targsrc  targsrc:lib_mysql/include/lib/lib_mysql.h  comment:""
 dev.targsrc  targsrc:lib_odbc/cpp/gen/lib_odbc_gen.cpp  comment:""
 dev.targsrc  targsrc:lib_odbc/cpp/lib/lib_odbc.cpp  comment:""
 dev.targsrc  targsrc:lib_odbc/include/gen/lib_odbc_gen.h  comment:""
@@ -326,10 +321,6 @@ dev.targsrc  targsrc:lib_sql/cpp/lib/lib_sql.cpp  comment:""
 dev.targsrc  targsrc:lib_sql/include/gen/lib_sql_gen.h  comment:""
 dev.targsrc  targsrc:lib_sql/include/gen/lib_sql_gen.inl.h  comment:""
 dev.targsrc  targsrc:lib_sql/include/lib_sql.h  comment:""
-dev.targsrc  targsrc:mysql2ssim/cpp/gen/mysql2ssim_gen.cpp  comment:""
-dev.targsrc  targsrc:mysql2ssim/cpp/mysql2ssim.cpp  comment:""
-dev.targsrc  targsrc:mysql2ssim/include/gen/mysql2ssim_gen.h  comment:""
-dev.targsrc  targsrc:mysql2ssim/include/gen/mysql2ssim_gen.inl.h  comment:""
 dev.targsrc  targsrc:src_func/cpp/gen/src_func_gen.cpp  comment:""
 dev.targsrc  targsrc:src_func/cpp/src/func/check.cpp  comment:"Check source code for common errors"
 dev.targsrc  targsrc:src_func/cpp/src/func/edit.cpp  comment:"Implementation of -e"
@@ -355,10 +346,6 @@ dev.targsrc  targsrc:ssim2csv/cpp/gen/ssim2csv_gen.cpp  comment:""
 dev.targsrc  targsrc:ssim2csv/cpp/ssim2csv.cpp  comment:""
 dev.targsrc  targsrc:ssim2csv/include/gen/ssim2csv_gen.h  comment:""
 dev.targsrc  targsrc:ssim2csv/include/gen/ssim2csv_gen.inl.h  comment:""
-dev.targsrc  targsrc:ssim2mysql/cpp/gen/ssim2mysql_gen.cpp  comment:""
-dev.targsrc  targsrc:ssim2mysql/cpp/ssim2mysql.cpp  comment:""
-dev.targsrc  targsrc:ssim2mysql/include/gen/ssim2mysql_gen.h  comment:""
-dev.targsrc  targsrc:ssim2mysql/include/gen/ssim2mysql_gen.inl.h  comment:""
 dev.targsrc  targsrc:strconv/cpp/gen/strconv_gen.cpp  comment:""
 dev.targsrc  targsrc:strconv/cpp/strconv/strconv.cpp  comment:""
 dev.targsrc  targsrc:strconv/include/gen/strconv_gen.h  comment:""
--- a/data/dev/targsyslib.ssim
+++ b/data/dev/targsyslib.ssim
@@ -3,4 +3,3 @@ dev.targsyslib  targsyslib:algo_lib.m  uname:Linux  comment:""
 dev.targsyslib  targsyslib:algo_lib.stdc++  uname:Linux  comment:""
 dev.targsyslib  targsyslib:amc.crypt  uname:Linux  comment:""
 dev.targsyslib  targsyslib:amc.crypto  uname:Linux  comment:""
-dev.targsyslib  targsyslib:lib_mysql.mysqlclient  uname:Linux  comment:""
--- a/data/dev/tool_opt.ssim
+++ b/data/dev/tool_opt.ssim
@@ -30,8 +30,6 @@ dev.tool_opt  tool_opt:185  opt_type:CC_OPTS  opt:-D_COVERAGE  target:""  uname:
 dev.tool_opt  tool_opt:218  opt_type:INCL  opt:.  target:""  uname:""  compiler:""  cfg:""  arch:""  comment:""
 dev.tool_opt  tool_opt:221  opt_type:CC_OPTS  opt:-msse4.2  target:""  uname:""  compiler:""  cfg:""  arch:x86_64  comment:""
 dev.tool_opt  tool_opt:222  opt_type:CC_OPTS  opt:"-D AOS_SSE42"  target:""  uname:""  compiler:""  cfg:""  arch:x86_64  comment:""
-dev.tool_opt  tool_opt:303  opt_type:LINK_OPTS  opt:"-L /usr/lib64/mysql"  target:mysql2ssim  uname:""  compiler:""  cfg:""  arch:""  comment:""
-dev.tool_opt  tool_opt:313  opt_type:LINK_OPTS  opt:"-L /usr/lib64/mysql"  target:ssim2mysql  uname:""  compiler:""  cfg:""  arch:""  comment:""
 dev.tool_opt  tool_opt:398  opt_type:CC_OPTS  opt:-Wno-vla  target:""  uname:""  compiler:""  cfg:""  arch:""  comment:"I hate this warning"
 dev.tool_opt  tool_opt:472  opt_type:CC_OPTS  opt:-Wno-unused-local-typedefs  target:""  uname:Linux  compiler:g++  cfg:""  arch:""  comment:""
 dev.tool_opt  tool_opt:476  opt_type:CC_OPTS  opt:-Wno-long-long  target:""  uname:Linux  compiler:""  cfg:""  arch:""  comment:""
--- a/data/dmmeta/anonfld.ssim
+++ b/data/dmmeta/anonfld.ssim
@@ -63,8 +63,6 @@ dmmeta.anonfld  field:command.amc_vis.ctype  comment:""
 dmmeta.anonfld  field:command.atf_amc.amctest  comment:""
 dmmeta.anonfld  field:command.atf_norm.normcheck  comment:""
 dmmeta.anonfld  field:command.atf_unit.unittest  comment:""
-dmmeta.anonfld  field:command.mysql2ssim.tables  comment:""
-dmmeta.anonfld  field:command.mysql2ssim.url  comment:""
 dmmeta.anonfld  field:command.src_func.body  comment:""
 dmmeta.anonfld  field:command.src_func.name  comment:""
 dmmeta.anonfld  field:command.src_func.targsrc  comment:""
--- a/data/dmmeta/cfmt.ssim
+++ b/data/dmmeta/cfmt.ssim
@@ -311,14 +311,12 @@ dmmeta.cfmt  cfmt:command.atf_norm.Argv  printfmt:Tuple  read:Y  print:Y  sep:""
 dmmeta.cfmt  cfmt:command.atf_unit.Argv  printfmt:Auto  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.lib_ctype.Argv  printfmt:Tuple  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.lib_exec.Argv  printfmt:Auto  read:Y  print:Y  sep:""  genop:N  comment:""
-dmmeta.cfmt  cfmt:command.mysql2ssim.Argv  printfmt:Auto  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.src_func.Argv  printfmt:Tuple  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.src_func.Tuple  printfmt:Auto  read:Y  print:N  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.src_hdr.Argv  printfmt:Tuple  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.src_lim.Argv  printfmt:Tuple  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.src_lim.Tuple  printfmt:Auto  read:Y  print:N  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.ssim2csv.Argv  printfmt:Auto  read:Y  print:Y  sep:""  genop:N  comment:""
-dmmeta.cfmt  cfmt:command.ssim2mysql.Argv  printfmt:Auto  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:command.strconv.Argv  printfmt:Tuple  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:dev.Arch.String  printfmt:Tuple  read:Y  print:Y  sep:""  genop:N  comment:""
 dmmeta.cfmt  cfmt:dev.Badline.String  printfmt:Tuple  read:Y  print:Y  sep:""  genop:Y  comment:""
@@ -501,15 +499,6 @@ dmmeta.cfmt  cfmt:report.src_func.String  printfmt:Tuple  read:Y  print:Y  sep:"
 dmmeta.cfmt  cfmt:ssim2csv.FExpand.String  printfmt:Raw  read:N  print:Y  sep:""  genop:Y  comment:""
 dmmeta.cfmt  cfmt:ssim2csv.FFlatten.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
 dmmeta.cfmt  cfmt:ssim2csv.FOutfile.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FCmd.String  printfmt:Raw  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FColumn.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FCtype.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FField.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FInput.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FNs.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FSqltype.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FSsimfile.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
-dmmeta.cfmt  cfmt:ssim2mysql.FSubstr.String  printfmt:Tuple  read:N  print:Y  sep:""  genop:Y  comment:""
 dmmeta.cfmt  cfmt:u128.String  printfmt:Extern  read:Y  print:Y  sep:""  genop:Y  comment:""
 dmmeta.cfmt  cfmt:u16.Json  printfmt:Extern  read:N  print:Y  sep:""  genop:Y  comment:""
 dmmeta.cfmt  cfmt:u16.String  printfmt:Extern  read:Y  print:Y  sep:""  genop:Y  comment:""
--- a/data/dmmeta/ctype.ssim
+++ b/data/dmmeta/ctype.ssim
@@ -572,12 +572,10 @@ dmmeta.ctype  ctype:command.atf_norm  comment:""
 dmmeta.ctype  ctype:command.atf_unit  comment:""
 dmmeta.ctype  ctype:command.lib_ctype  comment:""
 dmmeta.ctype  ctype:command.lib_exec  comment:""
-dmmeta.ctype  ctype:command.mysql2ssim  comment:""
 dmmeta.ctype  ctype:command.src_func  comment:""
 dmmeta.ctype  ctype:command.src_hdr  comment:""
 dmmeta.ctype  ctype:command.src_lim  comment:""
 dmmeta.ctype  ctype:command.ssim2csv  comment:""
-dmmeta.ctype  ctype:command.ssim2mysql  comment:""
 dmmeta.ctype  ctype:command.strconv  comment:""
 dmmeta.ctype  ctype:dev.Arch  comment:""
 dmmeta.ctype  ctype:dev.Badline  comment:""
@@ -774,8 +772,6 @@ dmmeta.ctype  ctype:lib_json.FDb  comment:""
 dmmeta.ctype  ctype:lib_json.FNode  comment:""
 dmmeta.ctype  ctype:lib_json.FParser  comment:""
 dmmeta.ctype  ctype:lib_json.FldKey  comment:""
-dmmeta.ctype  ctype:lib_mysql.FDb  comment:""
-dmmeta.ctype  ctype:lib_mysql.Res  comment:""
 dmmeta.ctype  ctype:lib_odbc.Datatype  comment:"Is this the right namespace?"
 dmmeta.ctype  ctype:lib_odbc.FDb  comment:""
 dmmeta.ctype  ctype:lib_odbc.Hstmt  comment:""
@@ -793,8 +789,6 @@ dmmeta.ctype  ctype:lin.Numamemfld  comment:""
 dmmeta.ctype  ctype:lin.Procstatusfld  comment:""
 dmmeta.ctype  ctype:lin.cap_data_t  comment:""
 dmmeta.ctype  ctype:lin.cap_header_t  comment:""
-dmmeta.ctype  ctype:mysql2ssim.FDb  comment:""
-dmmeta.ctype  ctype:mysql2ssim.FTobltin  comment:""
 dmmeta.ctype  ctype:pad_byte  comment:""
 dmmeta.ctype  ctype:pcap.hdr  comment:""
 dmmeta.ctype  ctype:pcap.rec_hdr  comment:""
@@ -836,16 +830,6 @@ dmmeta.ctype  ctype:ssim2csv.FDb  comment:""
 dmmeta.ctype  ctype:ssim2csv.FExpand  comment:""
 dmmeta.ctype  ctype:ssim2csv.FFlatten  comment:""
 dmmeta.ctype  ctype:ssim2csv.FOutfile  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FCmd  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FColumn  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FCtype  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FDb  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FField  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FInput  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FNs  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FSqltype  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FSsimfile  comment:""
-dmmeta.ctype  ctype:ssim2mysql.FSubstr  comment:""
 dmmeta.ctype  ctype:strconv.FDb  comment:""
 dmmeta.ctype  ctype:u128  comment:""
 dmmeta.ctype  ctype:u16  comment:""
--- a/data/dmmeta/ctypelen.ssim
+++ b/data/dmmeta/ctypelen.ssim
@@ -534,12 +534,10 @@ dmmeta.ctypelen  ctype:command.atf_norm  len:112  alignment:8  padbytes:0
 dmmeta.ctypelen  ctype:command.atf_unit  len:152  alignment:8  padbytes:14
 dmmeta.ctypelen  ctype:command.lib_ctype  len:16  alignment:8  padbytes:0
 dmmeta.ctypelen  ctype:command.lib_exec  len:12  alignment:4  padbytes:5
-dmmeta.ctypelen  ctype:command.mysql2ssim  len:72  alignment:8  padbytes:19
 dmmeta.ctypelen  ctype:command.src_func  len:712  alignment:8  padbytes:2
 dmmeta.ctypelen  ctype:command.src_hdr  len:120  alignment:8  padbytes:6
 dmmeta.ctypelen  ctype:command.src_lim  len:128  alignment:8  padbytes:12
 dmmeta.ctypelen  ctype:command.ssim2csv  len:24  alignment:8  padbytes:7
-dmmeta.ctypelen  ctype:command.ssim2mysql  len:80  alignment:8  padbytes:6
 dmmeta.ctypelen  ctype:command.strconv  len:40  alignment:8  padbytes:6
 dmmeta.ctypelen  ctype:dev.Arch  len:204  alignment:1  padbytes:0
 dmmeta.ctypelen  ctype:dev.Badline  len:458  alignment:1  padbytes:0
@@ -699,8 +697,6 @@ dmmeta.ctypelen  ctype:lib_json.FDb  len:296  alignment:8  padbytes:4
 dmmeta.ctypelen  ctype:lib_json.FNode  len:72  alignment:8  padbytes:11
 dmmeta.ctypelen  ctype:lib_json.FParser  len:96  alignment:8  padbytes:8
 dmmeta.ctypelen  ctype:lib_json.FldKey  len:24  alignment:8  padbytes:0
-dmmeta.ctypelen  ctype:lib_mysql.FDb  len:16  alignment:8  padbytes:7
-dmmeta.ctypelen  ctype:lib_mysql.Res  len:8  alignment:8  padbytes:0
 dmmeta.ctypelen  ctype:lib_odbc.Datatype  len:4  alignment:4  padbytes:0
 dmmeta.ctypelen  ctype:lib_odbc.Sqlfkey  len:660  alignment:1  padbytes:0
 dmmeta.ctypelen  ctype:lib_odbc.Sqlpkeycol  len:404  alignment:1  padbytes:0
@@ -713,8 +709,6 @@ dmmeta.ctypelen  ctype:lin.Limitsfld  len:100  alignment:1  padbytes:0
 dmmeta.ctypelen  ctype:lin.Numamemfld  len:50  alignment:1  padbytes:0
 dmmeta.ctypelen  ctype:lin.Procstatusfld  len:50  alignment:1  padbytes:0
 dmmeta.ctypelen  ctype:lin.cap_data_t  len:24  alignment:1  padbytes:0
-dmmeta.ctypelen  ctype:mysql2ssim.FDb  len:152  alignment:8  padbytes:7
-dmmeta.ctypelen  ctype:mysql2ssim.FTobltin  len:24  alignment:8  padbytes:6
 dmmeta.ctypelen  ctype:pcap.hdr  len:24  alignment:1  padbytes:0
 dmmeta.ctypelen  ctype:pcap.rec_hdr  len:16  alignment:1  padbytes:0
 dmmeta.ctypelen  ctype:report.Msgcount  len:192  alignment:4  padbytes:0
@@ -754,14 +748,4 @@ dmmeta.ctypelen  ctype:ssim2csv.FDb  len:640  alignment:8  padbytes:15
 dmmeta.ctypelen  ctype:ssim2csv.FExpand  len:24  alignment:8  padbytes:0
 dmmeta.ctypelen  ctype:ssim2csv.FFlatten  len:32  alignment:8  padbytes:0
 dmmeta.ctypelen  ctype:ssim2csv.FOutfile  len:48  alignment:8  padbytes:4
-dmmeta.ctypelen  ctype:ssim2mysql.FCmd  len:32  alignment:8  padbytes:0
-dmmeta.ctypelen  ctype:ssim2mysql.FColumn  len:144  alignment:8  padbytes:7
-dmmeta.ctypelen  ctype:ssim2mysql.FCtype  len:256  alignment:8  padbytes:4
-dmmeta.ctypelen  ctype:ssim2mysql.FDb  len:2120  alignment:8  padbytes:24
-dmmeta.ctypelen  ctype:ssim2mysql.FField  len:656  alignment:8  padbytes:11
-dmmeta.ctypelen  ctype:ssim2mysql.FInput  len:65704  alignment:8  padbytes:17
-dmmeta.ctypelen  ctype:ssim2mysql.FNs  len:248  alignment:8  padbytes:2
-dmmeta.ctypelen  ctype:ssim2mysql.FSqltype  len:306  alignment:1  padbytes:0
-dmmeta.ctypelen  ctype:ssim2mysql.FSsimfile  len:168  alignment:8  padbytes:14
-dmmeta.ctypelen  ctype:ssim2mysql.FSubstr  len:464  alignment:8  padbytes:0
 dmmeta.ctypelen  ctype:strconv.FDb  len:48  alignment:8  padbytes:7
--- a/data/dmmeta/dispsig.ssim
+++ b/data/dmmeta/dispsig.ssim
@@ -16,4 +16,3 @@ dmmeta.dispsig  dispsig:lib_ctype.Input  signature:ab81c986e2e85d3e267eb18d62976
 dmmeta.dispsig  dispsig:src_func.Input  signature:ed05427e5503605f81d52ef70aa44d1697440397
 dmmeta.dispsig  dispsig:src_hdr.Input  signature:bc0d709d29ce716f6afd2fa3b51b24f74ff6108e
 dmmeta.dispsig  dispsig:src_lim.Input  signature:4d68024a255bb013457d44262892315524185d24
-dmmeta.dispsig  dispsig:ssim2mysql.Input  signature:7abba95113c0af13de74ea1e2904b75056b22891
--- a/data/dmmeta/fbuf.ssim
+++ b/data/dmmeta/fbuf.ssim
@@ -1,3 +1,2 @@
 dmmeta.fbuf  field:atf_amc.Bytebuf.in  max:64  fbuftype:Bytebuf  insready:atf_amc.Bytebuf.in  inseof:atf_amc.Bytebuf.in  comment:""
 dmmeta.fbuf  field:atf_amc.Linebuf.in  max:64  fbuftype:Linebuf  insready:atf_amc.Linebuf.in  inseof:atf_amc.Linebuf.in  comment:""
-dmmeta.fbuf  field:ssim2mysql.FInput.in_buf  max:65536  fbuftype:Linebuf  insready:ssim2mysql.FDb.cd_input_line  inseof:ssim2mysql.FInput.in_buf  comment:""
--- a/data/dmmeta/fcleanup.ssim
+++ b/data/dmmeta/fcleanup.ssim
@@ -11,8 +11,6 @@ dmmeta.fcleanup  field:atf_amc.AmcCleanup2.field2  comment:""
 dmmeta.fcleanup  field:atf_amc.FCascdel.key  comment:""
 dmmeta.fcleanup  field:lib_iconv.Icd.icd  comment:""
 dmmeta.fcleanup  field:lib_json.FParser.root_node  comment:""
-dmmeta.fcleanup  field:lib_mysql.FDb.mysql  comment:""
-dmmeta.fcleanup  field:lib_mysql.Res.res  comment:""
 dmmeta.fcleanup  field:lib_odbc.FDb.hdbc  comment:""
 dmmeta.fcleanup  field:lib_odbc.FDb.henv  comment:""
 dmmeta.fcleanup  field:lib_odbc.Hstmt.hstmt  comment:""
--- a/data/dmmeta/fcmdline.ssim
+++ b/data/dmmeta/fcmdline.ssim
@@ -12,10 +12,8 @@ dmmeta.fcmdline  field:atf_amc.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:atf_norm.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:atf_unit.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:lib_exec.FDb.cmdline  read:Y  comment:""
-dmmeta.fcmdline  field:mysql2ssim.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:src_func.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:src_hdr.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:src_lim.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:ssim2csv.FDb.cmdline  read:Y  comment:""
-dmmeta.fcmdline  field:ssim2mysql.FDb.cmdline  read:Y  comment:""
 dmmeta.fcmdline  field:strconv.FDb.cmdline  read:Y  comment:""
--- a/data/dmmeta/field.ssim
+++ b/data/dmmeta/field.ssim
@@ -2083,14 +2083,6 @@ dmmeta.field  field:command.lib_exec.dry_run  arg:bool  reftype:Val  dflt:""  co
 dmmeta.field  field:command.lib_exec.q  arg:bool  reftype:Val  dflt:true  comment:"Do not print node name"
 dmmeta.field  field:command.lib_exec.maxjobs  arg:i32  reftype:Val  dflt:8  comment:"Maximum number of parallel jobs"
 dmmeta.field  field:command.lib_exec.complooo  arg:bool  reftype:Val  dflt:false  comment:""
-dmmeta.field  field:command.mysql2ssim.writessimfile  arg:bool  reftype:Val  dflt:""  comment:"Write to ssimfile directly"
-dmmeta.field  field:command.mysql2ssim.url  arg:algo.cstring  reftype:Val  dflt:""  comment:"user:pass@host/db or sock:///filename/db"
-dmmeta.field  field:command.mysql2ssim.tables  arg:algo.cstring  reftype:Val  dflt:'""'  comment:"comma-separated list of tables. Default is all tables"
-dmmeta.field  field:command.mysql2ssim.schema  arg:bool  reftype:Val  dflt:""  comment:"Generate ssim type definition"
-dmmeta.field  field:command.mysql2ssim.in  arg:algo.cstring  reftype:Val  dflt:'"data"'  comment:"Input directory or filename, - for stdin"
-dmmeta.field  field:command.mysql2ssim.pretty  arg:bool  reftype:Val  dflt:false  comment:"Format output for the screen"
-dmmeta.field  field:command.mysql2ssim.nologo  arg:bool  reftype:Val  dflt:""  comment:"Don't show copyright notice"
-dmmeta.field  field:command.mysql2ssim.baddbok  arg:bool  reftype:Val  dflt:""  comment:"Don't claim if bad database"
 dmmeta.field  field:command.src_func.in  arg:algo.cstring  reftype:Val  dflt:'"data"'  comment:"Input directory or filename, - for stdin"
 dmmeta.field  field:command.src_func.targsrc  arg:dev.Targsrc  reftype:RegxSql  dflt:'"%"'  comment:"Visit these sources (accepts target name)"
 dmmeta.field  field:command.src_func.name  arg:algo.cstring  reftype:RegxSql  dflt:'"%"'  comment:"(with -listfunc) Match function name"
@@ -2122,17 +2114,6 @@ dmmeta.field  field:command.src_lim.capture  arg:bool  reftype:Val  dflt:""  com
 dmmeta.field  field:command.src_lim.write  arg:bool  reftype:Val  dflt:""  comment:"Update ssim database (with -capture)"
 dmmeta.field  field:command.ssim2csv.expand  arg:algo.cstring  reftype:Val  dflt:'""'  comment:""
 dmmeta.field  field:command.ssim2csv.ignoreQuote  arg:bool  reftype:Val  dflt:false  comment:""
-dmmeta.field  field:command.ssim2mysql.url  arg:algo.cstring  reftype:Val  dflt:'""'  comment:"URL of mysql server. user:pass@hostb or sock://filename; Empty -> stdout"
-dmmeta.field  field:command.ssim2mysql.data_dir  arg:algo.cstring  reftype:Val  dflt:'"data"'  comment:"Load dmmeta info from this directory"
-dmmeta.field  field:command.ssim2mysql.maxpacket  arg:i32  reftype:Val  dflt:100000  comment:"Max Mysql packet size"
-dmmeta.field  field:command.ssim2mysql.replace  arg:bool  reftype:Val  dflt:true  comment:"use REPLACE INTO instead of INSERT INTO"
-dmmeta.field  field:command.ssim2mysql.trunc  arg:bool  reftype:Val  dflt:""  comment:"Truncate target table"
-dmmeta.field  field:command.ssim2mysql.dry_run  arg:bool  reftype:Val  dflt:""  comment:"Print SQL commands to the stdout"
-dmmeta.field  field:command.ssim2mysql.fldfunc  arg:bool  reftype:Val  dflt:false  comment:"create columns for fldfuncs"
-dmmeta.field  field:command.ssim2mysql.in  arg:algo.cstring  reftype:Val  dflt:'"-"'  comment:"Input directory or filename, - for stdin"
-dmmeta.field  field:command.ssim2mysql.db  arg:algo.cstring  reftype:Val  dflt:'""'  comment:"Optional database name"
-dmmeta.field  field:command.ssim2mysql.createdb  arg:bool  reftype:Val  dflt:""  comment:"Emit CREATE DATABASE code for namespace specified with <db>"
-dmmeta.field  field:command.ssim2mysql.fkey  arg:bool  reftype:Val  dflt:""  comment:"Enable foreign key constraints (uses InnoDB storage engine)"
 dmmeta.field  field:command.strconv.str  arg:algo.cstring  reftype:Val  dflt:""  comment:"String parameter"
 dmmeta.field  field:command.strconv.tocamelcase  arg:bool  reftype:Val  dflt:false  comment:"Convert string to camel case"
 dmmeta.field  field:command.strconv.tolowerunder  arg:bool  reftype:Val  dflt:false  comment:"Convert string to lower-under"
@@ -2733,10 +2714,6 @@ dmmeta.field  field:lib_json.FParser.uesc_need  arg:u8  reftype:Val  dflt:""  co
 dmmeta.field  field:lib_json.FParser.value  arg:algo.cstring  reftype:Val  dflt:""  comment:""
 dmmeta.field  field:lib_json.FldKey.object  arg:lib_json.FNode  reftype:Ptr  dflt:""  comment:""
 dmmeta.field  field:lib_json.FldKey.field  arg:algo.strptr  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:lib_mysql.FDb._db  arg:lib_mysql.FDb  reftype:Global  dflt:""  comment:""
-dmmeta.field  field:lib_mysql.FDb.mysql  arg:MYSQL  reftype:Ptr  dflt:""  comment:""
-dmmeta.field  field:lib_mysql.FDb.res  arg:lib_mysql.Res  reftype:Cppstack  dflt:""  comment:""
-dmmeta.field  field:lib_mysql.Res.res  arg:MYSQL_RES  reftype:Ptr  dflt:""  comment:""
 dmmeta.field  field:lib_odbc.Datatype.datatype  arg:u32  reftype:Val  dflt:""  comment:""
 dmmeta.field  field:lib_odbc.FDb.henv  arg:SQLHANDLE  reftype:Val  dflt:SQL_NULL_HANDLE  comment:"Environment handle"
 dmmeta.field  field:lib_odbc.FDb.hdbc  arg:SQLHANDLE  reftype:Val  dflt:SQL_NULL_HANDLE  comment:"DB Connection handle"
@@ -2780,14 +2757,6 @@ dmmeta.field  field:lin.cap_data_t.permitted  arg:u64  reftype:Val  dflt:""  com
 dmmeta.field  field:lin.cap_data_t.inheritable  arg:u64  reftype:Val  dflt:""  comment:""
 dmmeta.field  field:lin.cap_header_t.version  arg:u32  reftype:Val  dflt:""  comment:""
 dmmeta.field  field:lin.cap_header_t.pid  arg:pid_t  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:mysql2ssim.FDb._db  arg:mysql2ssim.FDb  reftype:Global  dflt:""  comment:""
-dmmeta.field  field:mysql2ssim.FDb.table_names  arg:algo.cstring  reftype:Tary  dflt:""  comment:"List of all tables obtained from mysql"
-dmmeta.field  field:mysql2ssim.FDb.in_tables  arg:algo.cstring  reftype:Tary  dflt:""  comment:""
-dmmeta.field  field:mysql2ssim.FDb.cmdline  arg:command.mysql2ssim  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:mysql2ssim.FDb.tempfile  arg:algo_lib.FTempfile  reftype:Val  dflt:""  comment:"Temp file where downloaded mysql goes before being handed to acr"
-dmmeta.field  field:mysql2ssim.FTobltin.warn  arg:bool  reftype:Val  dflt:""  comment:"Conversion warning"
-dmmeta.field  field:mysql2ssim.FTobltin.err  arg:bool  reftype:Val  dflt:""  comment:"Conversion error"
-dmmeta.field  field:mysql2ssim.FTobltin.vals  arg:algo.cstring  reftype:Tary  dflt:""  comment:"Used during schema extraction"
 dmmeta.field  field:pad_byte.value  arg:u8  reftype:Val  dflt:0  comment:""
 dmmeta.field  field:pcap.hdr.magic_number  arg:u32  reftype:Val  dflt:""  comment:"magic number"
 dmmeta.field  field:pcap.hdr.version_major  arg:u16  reftype:Val  dflt:""  comment:"major version number"
@@ -2995,58 +2964,5 @@ dmmeta.field  field:ssim2csv.FFlatten.prefix  arg:algo.cstring  reftype:Val  dfl
 dmmeta.field  field:ssim2csv.FOutfile.fildes  arg:algo_lib.FFildes  reftype:Val  dflt:""  comment:""
 dmmeta.field  field:ssim2csv.FOutfile.header  arg:algo.cstring  reftype:Val  dflt:""  comment:""
 dmmeta.field  field:ssim2csv.FOutfile.outfile  arg:algo.cstring  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FCmd.cmd  arg:algo.cstring  reftype:Val  dflt:""  comment:"Command to execute"
-dmmeta.field  field:ssim2mysql.FColumn.column  arg:algo.Smallstr100  reftype:Val  dflt:""  comment:"Column name"
-dmmeta.field  field:ssim2mysql.FColumn.name  arg:algo.Smallstr50  reftype:Val  dflt:""  comment:"Column name"
-dmmeta.field  field:ssim2mysql.FColumn.defval  arg:algo.cstring  reftype:Val  dflt:""  comment:"Default value (sql expression)"
-dmmeta.field  field:ssim2mysql.FColumn.p_field  arg:ssim2mysql.FField  reftype:Upptr  dflt:""  comment:"Associated field"
-dmmeta.field  field:ssim2mysql.FColumn.is_int  arg:bool  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FColumn.ssimfile  arg:dmmeta.Ssimfile  reftype:Pkey  dflt:""  comment:"Associated ssimfile"
-dmmeta.field  field:ssim2mysql.FColumn.is_pkey  arg:bool  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FCtype.msghdr  arg:dmmeta.Ctype  reftype:Base  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FCtype.p_ns  arg:ssim2mysql.FNs  reftype:Upptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FCtype.c_sqltype  arg:ssim2mysql.FSqltype  reftype:Ptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FCtype.c_field  arg:ssim2mysql.FField  reftype:Ptrary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FCtype.c_ssimfile  arg:ssim2mysql.FSsimfile  reftype:Ptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ind_column  arg:ssim2mysql.FColumn  reftype:Thash  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ns  arg:ssim2mysql.FNs  reftype:Lary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.cmdline  arg:command.ssim2mysql  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ctype  arg:ssim2mysql.FCtype  reftype:Lary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.field  arg:ssim2mysql.FField  reftype:Lary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.sqltype  arg:ssim2mysql.FSqltype  reftype:Lary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.cmd  arg:ssim2mysql.FCmd  reftype:Tpool  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.column  arg:ssim2mysql.FColumn  reftype:Lary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb._db  arg:ssim2mysql.FDb  reftype:Global  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ind_ns  arg:ssim2mysql.FNs  reftype:Thash  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ind_ctype  arg:ssim2mysql.FCtype  reftype:Thash  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ind_field  arg:ssim2mysql.FField  reftype:Thash  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.substr  arg:ssim2mysql.FSubstr  reftype:Lary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ssimfile  arg:ssim2mysql.FSsimfile  reftype:Lary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.ind_ssimfile  arg:ssim2mysql.FSsimfile  reftype:Thash  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.zs_cmd  arg:ssim2mysql.FCmd  reftype:Llist  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.c_cmd_cur  arg:ssim2mysql.FCmd  reftype:Ptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.c_ssimfile_cur  arg:ssim2mysql.FSsimfile  reftype:Ptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.n_cmd_rows  arg:u32  reftype:Val  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.input  arg:ssim2mysql.FInput  reftype:Tpool  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.zd_ssimfile  arg:ssim2mysql.FSsimfile  reftype:Llist  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FDb.cd_input_line  arg:ssim2mysql.FInput  reftype:Llist  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FField.msghdr  arg:dmmeta.Field  reftype:Base  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FField.p_ctype  arg:ssim2mysql.FCtype  reftype:Upptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FField.p_arg  arg:ssim2mysql.FCtype  reftype:Upptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FField.c_substr  arg:ssim2mysql.FSubstr  reftype:Ptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FField.select  arg:bool  reftype:Val  dflt:""  comment:"Select for inclusion to upload"
-dmmeta.field  field:ssim2mysql.FField.is_pkeyref  arg:bool  reftype:Val  dflt:""  comment:"Field is a pkey reference to something else"
-dmmeta.field  field:ssim2mysql.FInput.in_buf  arg:char  reftype:Fbuf  dflt:"'\\n'"  comment:"Socket data comes here"
-dmmeta.field  field:ssim2mysql.FInput.tuple  arg:algo.Tuple  reftype:Val  dflt:""  comment:"Temp tuple"
-dmmeta.field  field:ssim2mysql.FInput.warnstr  arg:algo.cstring  reftype:Val  dflt:""  comment:"Warning string"
-dmmeta.field  field:ssim2mysql.FNs.msghdr  arg:dmmeta.Ns  reftype:Base  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FNs.c_ssimfile  arg:ssim2mysql.FSsimfile  reftype:Ptrary  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FSqltype.msghdr  arg:dmmeta.Sqltype  reftype:Base  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FSsimfile.msghdr  arg:dmmeta.Ssimfile  reftype:Base  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FSsimfile.p_ctype  arg:ssim2mysql.FCtype  reftype:Upptr  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FSsimfile.truncated  arg:bool  reftype:Val  dflt:""  comment:"Already truncated in database"
-dmmeta.field  field:ssim2mysql.FSsimfile.c_column  arg:ssim2mysql.FColumn  reftype:Ptrary  dflt:""  comment:"List of columns"
-dmmeta.field  field:ssim2mysql.FSubstr.msghdr  arg:dmmeta.Substr  reftype:Base  dflt:""  comment:""
-dmmeta.field  field:ssim2mysql.FSubstr.p_field  arg:ssim2mysql.FField  reftype:Upptr  dflt:""  comment:""
 dmmeta.field  field:strconv.FDb._db  arg:strconv.FDb  reftype:Global  dflt:""  comment:""
 dmmeta.field  field:strconv.FDb.cmdline  arg:command.strconv  reftype:Val  dflt:""  comment:""
--- a/data/dmmeta/finput.ssim
+++ b/data/dmmeta/finput.ssim
@@ -171,9 +171,3 @@ dmmeta.finput  field:src_lim.FDb.gitfile  extrn:N  update:N  strict:N  comment:"
 dmmeta.finput  field:src_lim.FDb.include  extrn:N  update:N  strict:N  comment:""
 dmmeta.finput  field:src_lim.FDb.linelim  extrn:N  update:N  strict:N  comment:""
 dmmeta.finput  field:src_lim.FDb.targsrc  extrn:N  update:N  strict:N  comment:""
-dmmeta.finput  field:ssim2mysql.FDb.ctype  extrn:N  update:N  strict:Y  comment:""
-dmmeta.finput  field:ssim2mysql.FDb.field  extrn:N  update:N  strict:Y  comment:""
-dmmeta.finput  field:ssim2mysql.FDb.ns  extrn:N  update:N  strict:Y  comment:""
-dmmeta.finput  field:ssim2mysql.FDb.sqltype  extrn:N  update:N  strict:Y  comment:""
-dmmeta.finput  field:ssim2mysql.FDb.ssimfile  extrn:N  update:N  strict:Y  comment:""
-dmmeta.finput  field:ssim2mysql.FDb.substr  extrn:N  update:N  strict:Y  comment:""
--- a/data/dmmeta/floadtuples.ssim
+++ b/data/dmmeta/floadtuples.ssim
@@ -13,5 +13,4 @@ dmmeta.floadtuples  field:command.lib_ctype.in  comment:""
 dmmeta.floadtuples  field:command.src_func.in  comment:""
 dmmeta.floadtuples  field:command.src_hdr.in  comment:""
 dmmeta.floadtuples  field:command.src_lim.in  comment:""
-dmmeta.floadtuples  field:command.ssim2mysql.data_dir  comment:""
 dmmeta.floadtuples  field:command.strconv.in  comment:""
--- a/data/dmmeta/fnoremove.ssim
+++ b/data/dmmeta/fnoremove.ssim
@@ -151,9 +151,3 @@ dmmeta.fnoremove  field:src_lim.FDb.gitfile  comment:""
 dmmeta.fnoremove  field:src_lim.FDb.include  comment:""
 dmmeta.fnoremove  field:src_lim.FDb.linelim  comment:""
 dmmeta.fnoremove  field:src_lim.FDb.targsrc  comment:""
-dmmeta.fnoremove  field:ssim2mysql.FDb.ctype  comment:""
-dmmeta.fnoremove  field:ssim2mysql.FDb.field  comment:""
-dmmeta.fnoremove  field:ssim2mysql.FDb.ns  comment:""
-dmmeta.fnoremove  field:ssim2mysql.FDb.sqltype  comment:""
-dmmeta.fnoremove  field:ssim2mysql.FDb.ssimfile  comment:""
-dmmeta.fnoremove  field:ssim2mysql.FDb.substr  comment:""
--- a/data/dmmeta/fstep.ssim
+++ b/data/dmmeta/fstep.ssim
@@ -6,5 +6,3 @@ dmmeta.fstep  fstep:atf_amc.FDb.zs_t_typec  steptype:Extern  comment:""
 dmmeta.fstep  fstep:atf_amc.FDb.csl_h_typec  steptype:Extern  comment:""
 dmmeta.fstep  fstep:atf_amc.FDb.cs_t_typec  steptype:Extern  comment:""
 dmmeta.fstep  fstep:atf_amc.FDb.cd_in_msg  steptype:Inline  comment:""
-dmmeta.fstep  fstep:ssim2mysql.FDb.zs_cmd  steptype:Inline  comment:""
-dmmeta.fstep  fstep:ssim2mysql.FDb.cd_input_line  steptype:Inline  comment:""
--- a/data/dmmeta/ftrace.ssim
+++ b/data/dmmeta/ftrace.ssim
@@ -9,5 +9,3 @@ dmmeta.ftrace  field:atf_amc.FDb.csl_h_typec  comment:""
 dmmeta.ftrace  field:atf_amc.FDb.zs_t_typec  comment:""
 dmmeta.ftrace  field:atf_amc.FDb.zsl_h_typec  comment:""
 dmmeta.ftrace  field:lib_json.FDb.node  comment:""
-dmmeta.ftrace  field:ssim2mysql.FDb.cd_input_line  comment:""
-dmmeta.ftrace  field:ssim2mysql.FDb.zs_cmd  comment:""
--- a/data/dmmeta/llist.ssim
+++ b/data/dmmeta/llist.ssim
@@ -61,6 +61,3 @@ dmmeta.llist  field:lib_exec.FDb.zd_started  havetail:Y  havecount:Y  comment:""
 dmmeta.llist  field:src_func.FTarget.cd_targsrc  havetail:N  havecount:Y  comment:""
 dmmeta.llist  field:src_func.FTargsrc.zd_func  havetail:Y  havecount:Y  comment:""
 dmmeta.llist  field:src_lim.FGitfile.zd_include  havetail:Y  havecount:Y  comment:""
-dmmeta.llist  field:ssim2mysql.FDb.cd_input_line  havetail:N  havecount:N  comment:""
-dmmeta.llist  field:ssim2mysql.FDb.zd_ssimfile  havetail:Y  havecount:N  comment:""
-dmmeta.llist  field:ssim2mysql.FDb.zs_cmd  havetail:Y  havecount:N  comment:""
--- a/data/dmmeta/main.ssim
+++ b/data/dmmeta/main.ssim
@@ -13,10 +13,8 @@ dmmeta.main  ns:atf_unit  ismodule:N
 dmmeta.main  ns:lib_ctype  ismodule:N
 dmmeta.main  ns:lib_git  ismodule:N
 dmmeta.main  ns:lib_sql  ismodule:N
-dmmeta.main  ns:mysql2ssim  ismodule:N
 dmmeta.main  ns:src_func  ismodule:N
 dmmeta.main  ns:src_hdr  ismodule:N
 dmmeta.main  ns:src_lim  ismodule:N
 dmmeta.main  ns:ssim2csv  ismodule:N
-dmmeta.main  ns:ssim2mysql  ismodule:N
 dmmeta.main  ns:strconv  ismodule:N
--- a/data/dmmeta/ns.ssim
+++ b/data/dmmeta/ns.ssim
@@ -29,16 +29,13 @@ dmmeta.ns  ns:lib_exec  nstype:lib  comment:"Build and run a dag of subprocesses
 dmmeta.ns  ns:lib_git  nstype:lib  comment:""
 dmmeta.ns  ns:lib_iconv  nstype:lib  comment:"ICONV wrapper"
 dmmeta.ns  ns:lib_json  nstype:lib  comment:""
-dmmeta.ns  ns:lib_mysql  nstype:lib  comment:"Mysql adaptor"
 dmmeta.ns  ns:lib_odbc  nstype:lib  comment:"ODBC wrapper"
 dmmeta.ns  ns:lib_sql  nstype:lib  comment:"SQL formatting functions"
 dmmeta.ns  ns:lin  nstype:protocol  comment:"Linux API"
-dmmeta.ns  ns:mysql2ssim  nstype:exe  comment:"mysql -> ssim conversion tool"
 dmmeta.ns  ns:pcap  nstype:protocol  comment:""
 dmmeta.ns  ns:report  nstype:protocol  comment:"Report interfaces"
 dmmeta.ns  ns:src_func  nstype:exe  comment:"Access / edit functions"
 dmmeta.ns  ns:src_hdr  nstype:exe  comment:"Update source file / copyright header"
 dmmeta.ns  ns:src_lim  nstype:exe  comment:"Enforce line length, function length, and indentation limits"
 dmmeta.ns  ns:ssim2csv  nstype:exe  comment:"Ssim -> csv conversion tool"
-dmmeta.ns  ns:ssim2mysql  nstype:exe  comment:"Ssim -> mysql"
 dmmeta.ns  ns:strconv  nstype:exe  comment:""
--- a/data/dmmeta/nsinclude.ssim
+++ b/data/dmmeta/nsinclude.ssim
@@ -1,5 +1,4 @@
 dmmeta.nsinclude  nsinclude:atf_amc/include/lib/lib_json.h  sys:N  comment:""
 dmmeta.nsinclude  nsinclude:lib_iconv/iconv.h  sys:Y  comment:""
-dmmeta.nsinclude  nsinclude:lib_mysql/mysql/mysql.h  sys:Y  comment:""
 dmmeta.nsinclude  nsinclude:lib_odbc/sql.h  sys:Y  comment:""
 dmmeta.nsinclude  nsinclude:lib_odbc/sqlext.h  sys:Y  comment:""
--- a/data/dmmeta/nsversion.ssim
+++ b/data/dmmeta/nsversion.ssim
@@ -10,10 +10,8 @@ dmmeta.nsversion  ns:amc_vis  version:2.0.1  comment:""
 dmmeta.nsversion  ns:atf_amc  version:2.0.1  comment:""
 dmmeta.nsversion  ns:atf_norm  version:2.0.1  comment:""
 dmmeta.nsversion  ns:atf_unit  version:2.0.1  comment:""
-dmmeta.nsversion  ns:mysql2ssim  version:2.0.1  comment:""
 dmmeta.nsversion  ns:src_func  version:2.0.1  comment:""
 dmmeta.nsversion  ns:src_hdr  version:2.0.1  comment:""
 dmmeta.nsversion  ns:src_lim  version:2.0.1  comment:""
 dmmeta.nsversion  ns:ssim2csv  version:2.0.1  comment:""
-dmmeta.nsversion  ns:ssim2mysql  version:2.0.1  comment:""
 dmmeta.nsversion  ns:strconv  version:2.0.1  comment:""
--- a/data/dmmeta/nsx.ssim
+++ b/data/dmmeta/nsx.ssim
@@ -27,16 +27,13 @@ dmmeta.nsx  ns:lib_exec  genthrow:N  correct_getorcreate:Y  pool:algo_lib.FDb.ma
 dmmeta.nsx  ns:lib_git  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:Y  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:lib_iconv  genthrow:N  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:lib_json  genthrow:N  correct_getorcreate:Y  pool:lib_json.FDb.lpool  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
-dmmeta.nsx  ns:lib_mysql  genthrow:N  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:lib_odbc  genthrow:N  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:lib_sql  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:Y  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:lin  genthrow:N  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:Y  fldoffset_asserts:Y  comment:""
-dmmeta.nsx  ns:mysql2ssim  genthrow:N  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:pcap  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:Y  fldoffset_asserts:Y  comment:""
 dmmeta.nsx  ns:report  genthrow:N  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:Y  comment:""
 dmmeta.nsx  ns:src_func  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:Y  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:src_hdr  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:Y  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:src_lim  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:Y  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:ssim2csv  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
-dmmeta.nsx  ns:ssim2mysql  genthrow:Y  correct_getorcreate:N  pool:algo_lib.FDb.malloc  sortxref:N  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
 dmmeta.nsx  ns:strconv  genthrow:Y  correct_getorcreate:Y  pool:algo_lib.FDb.malloc  sortxref:Y  gbounsafe:N  pack:N  fldoffset_asserts:N  comment:""
--- a/data/dmmeta/ptrary.ssim
+++ b/data/dmmeta/ptrary.ssim
@@ -71,6 +71,3 @@ dmmeta.ptrary  field:lib_exec.FSyscmd.c_next  unique:Y
 dmmeta.ptrary  field:lib_exec.FSyscmd.c_prior  unique:Y
 dmmeta.ptrary  field:lib_json.FNode.c_child  unique:Y
 dmmeta.ptrary  field:src_hdr.FNs.c_targsrc  unique:Y
-dmmeta.ptrary  field:ssim2mysql.FCtype.c_field  unique:Y
-dmmeta.ptrary  field:ssim2mysql.FNs.c_ssimfile  unique:Y
-dmmeta.ptrary  field:ssim2mysql.FSsimfile.c_column  unique:Y
--- a/data/dmmeta/substr.ssim
+++ b/data/dmmeta/substr.ssim
@@ -83,5 +83,3 @@ dmmeta.substr  field:lib_odbc.Sqlpkeycol.column  expr:.RR  srcfield:lib_odbc.Sql
 dmmeta.substr  field:lib_odbc.Sqlpkeycol.pkey  expr:.LL  srcfield:lib_odbc.Sqlpkeycol.pkeycol
 dmmeta.substr  field:lib_odbc.Sqlukeycol.column  expr:.RR  srcfield:lib_odbc.Sqlukeycol.ukeycol
 dmmeta.substr  field:lib_odbc.Sqlukeycol.ukey  expr:.LL  srcfield:lib_odbc.Sqlukeycol.ukeycol
-dmmeta.substr  field:ssim2mysql.FColumn.name  expr:.RR  srcfield:ssim2mysql.FColumn.column
-dmmeta.substr  field:ssim2mysql.FColumn.ssimfile  expr:.RL  srcfield:ssim2mysql.FColumn.column
--- a/data/dmmeta/tary.ssim
+++ b/data/dmmeta/tary.ssim
@@ -30,9 +30,6 @@ dmmeta.tary  field:atf_unit.FPerfSort.orig  aliased:N  comment:""
 dmmeta.tary  field:atf_unit.FPerfSort.sorted  aliased:Y  comment:""
 dmmeta.tary  field:atf_unit.ShStreamAry.shstream  aliased:N  comment:""
 dmmeta.tary  field:lib_odbc.Sqlcharbuf.value  aliased:N  comment:""
-dmmeta.tary  field:mysql2ssim.FDb.in_tables  aliased:N  comment:""
-dmmeta.tary  field:mysql2ssim.FDb.table_names  aliased:N  comment:""
-dmmeta.tary  field:mysql2ssim.FTobltin.vals  aliased:N  comment:""
 dmmeta.tary  field:ssim2csv.FDb.flatten  aliased:N  comment:""
 dmmeta.tary  field:ssim2csv.FDb.name  aliased:N  comment:""
 dmmeta.tary  field:ssim2csv.FDb.value  aliased:N  comment:""
--- a/data/dmmeta/thash.ssim
+++ b/data/dmmeta/thash.ssim
@@ -111,8 +111,3 @@ dmmeta.thash  field:src_hdr.FDb.ind_ns  hashfld:dmmeta.Ns.ns  unique:Y  comment:
 dmmeta.thash  field:src_lim.FDb.ind_gitfile  hashfld:dev.Gitfile.gitfile  unique:Y  comment:""
 dmmeta.thash  field:ssim2csv.FDb.ind_expand  hashfld:ssim2csv.FExpand.expand  unique:Y  comment:""
 dmmeta.thash  field:ssim2csv.FDb.ind_outfile  hashfld:ssim2csv.FOutfile.outfile  unique:Y  comment:""
-dmmeta.thash  field:ssim2mysql.FDb.ind_column  hashfld:ssim2mysql.FColumn.column  unique:Y  comment:""
-dmmeta.thash  field:ssim2mysql.FDb.ind_ctype  hashfld:dmmeta.Ctype.ctype  unique:Y  comment:""
-dmmeta.thash  field:ssim2mysql.FDb.ind_field  hashfld:dmmeta.Field.field  unique:Y  comment:""
-dmmeta.thash  field:ssim2mysql.FDb.ind_ns  hashfld:dmmeta.Ns.ns  unique:Y  comment:""
-dmmeta.thash  field:ssim2mysql.FDb.ind_ssimfile  hashfld:dmmeta.Ssimfile.ssimfile  unique:Y  comment:""
--- a/data/dmmeta/tracefld.ssim
+++ b/data/dmmeta/tracefld.ssim
@@ -19,10 +19,6 @@ dmmeta.tracefld  tracefld:atf_amc.trace.step_cs_t_typec  comment:""
 dmmeta.tracefld  tracefld:atf_amc.trace.step_cs_t_typec_cycles  comment:""
 dmmeta.tracefld  tracefld:atf_amc.trace.step_cd_in_msg  comment:""
 dmmeta.tracefld  tracefld:atf_amc.trace.step_cd_in_msg_cycles  comment:""
-dmmeta.tracefld  tracefld:ssim2mysql.trace.step_zs_cmd  comment:""
-dmmeta.tracefld  tracefld:ssim2mysql.trace.step_zs_cmd_cycles  comment:""
-dmmeta.tracefld  tracefld:ssim2mysql.trace.step_cd_input_line  comment:""
-dmmeta.tracefld  tracefld:ssim2mysql.trace.step_cd_input_line_cycles  comment:""
 dmmeta.tracefld  tracefld:algo_lib.trace.alloc__db_malloc  comment:""
 dmmeta.tracefld  tracefld:algo_lib.trace.del__db_malloc  comment:""
 dmmeta.tracefld  tracefld:lib_json.trace.alloc__db_node  comment:""
--- a/data/dmmeta/tracerec.ssim
+++ b/data/dmmeta/tracerec.ssim
@@ -16,13 +16,10 @@ dmmeta.tracerec  tracerec:lib_exec.trace  comment:""
 dmmeta.tracerec  tracerec:lib_git.trace  comment:""
 dmmeta.tracerec  tracerec:lib_iconv.trace  comment:""
 dmmeta.tracerec  tracerec:lib_json.trace  comment:""
-dmmeta.tracerec  tracerec:lib_mysql.trace  comment:""
 dmmeta.tracerec  tracerec:lib_odbc.trace  comment:""
 dmmeta.tracerec  tracerec:lib_sql.trace  comment:""
-dmmeta.tracerec  tracerec:mysql2ssim.trace  comment:""
 dmmeta.tracerec  tracerec:src_func.trace  comment:""
 dmmeta.tracerec  tracerec:src_hdr.trace  comment:""
 dmmeta.tracerec  tracerec:src_lim.trace  comment:""
 dmmeta.tracerec  tracerec:ssim2csv.trace  comment:""
-dmmeta.tracerec  tracerec:ssim2mysql.trace  comment:""
 dmmeta.tracerec  tracerec:strconv.trace  comment:""
--- a/data/dmmeta/xref.ssim
+++ b/data/dmmeta/xref.ssim
@@ -438,22 +438,3 @@ dmmeta.xref  field:src_lim.FTargsrc.p_gitfile  inscond:true  via:src_lim.FDb.ind
 dmmeta.xref  field:src_lim.FGitfile.c_targsrc  inscond:true  via:src_lim.FDb.ind_gitfile/dev.Targsrc.src
 dmmeta.xref  field:ssim2csv.FDb.ind_expand  inscond:true  via:""
 dmmeta.xref  field:ssim2csv.FDb.ind_outfile  inscond:true  via:""
-dmmeta.xref  field:ssim2mysql.FCtype.p_ns  inscond:true  via:ssim2mysql.FDb.ind_ns/dmmeta.Ctype.ns
-dmmeta.xref  field:ssim2mysql.FCtype.c_sqltype  inscond:true  via:ssim2mysql.FDb.ind_ctype/dmmeta.Sqltype.ctype
-dmmeta.xref  field:ssim2mysql.FCtype.c_field  inscond:true  via:ssim2mysql.FDb.ind_ctype/dmmeta.Field.ctype
-dmmeta.xref  field:ssim2mysql.FDb.ind_column  inscond:true  via:""
-dmmeta.xref  field:ssim2mysql.FDb.ind_ns  inscond:true  via:""
-dmmeta.xref  field:ssim2mysql.FDb.ind_ctype  inscond:true  via:""
-dmmeta.xref  field:ssim2mysql.FDb.ind_field  inscond:true  via:""
-dmmeta.xref  field:ssim2mysql.FDb.ind_ssimfile  inscond:true  via:""
-dmmeta.xref  field:ssim2mysql.FDb.zs_cmd  inscond:true  via:""
-dmmeta.xref  field:ssim2mysql.FField.p_ctype  inscond:true  via:ssim2mysql.FDb.ind_ctype/dmmeta.Field.ctype
-dmmeta.xref  field:ssim2mysql.FField.p_arg  inscond:true  via:ssim2mysql.FDb.ind_ctype/dmmeta.Field.arg
-dmmeta.xref  field:ssim2mysql.FField.c_substr  inscond:true  via:ssim2mysql.FDb.ind_field/dmmeta.Substr.field
-dmmeta.xref  field:ssim2mysql.FNs.c_ssimfile  inscond:true  via:ssim2mysql.FDb.ind_ns/dmmeta.Ssimfile.ns
-dmmeta.xref  field:ssim2mysql.FSsimfile.p_ctype  inscond:true  via:ssim2mysql.FDb.ind_ctype/dmmeta.Ssimfile.ctype
-dmmeta.xref  field:ssim2mysql.FSsimfile.c_column  inscond:true  via:ssim2mysql.FDb.ind_ssimfile/ssim2mysql.FColumn.ssimfile
-dmmeta.xref  field:ssim2mysql.FSubstr.p_field  inscond:true  via:ssim2mysql.FDb.ind_field/dmmeta.Substr.field
-dmmeta.xref  field:ssim2mysql.FCtype.c_ssimfile  inscond:true  via:ssim2mysql.FDb.ind_ctype/dmmeta.Ssimfile.ctype
-dmmeta.xref  field:ssim2mysql.FDb.zd_ssimfile  inscond:false  via:""
-dmmeta.xref  field:ssim2mysql.FDb.cd_input_line  inscond:false  via:""
